<!DOCTYPE html><html><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width"/><link rel="icon" href="/SARS-Cov2/favicon.ico"/><title>Work_documented</title><meta name="robots" content="index,follow"/><meta name="googlebot" content="index,follow"/><meta name="description" content="This is file contains all the steps done for the master thesis"/><meta property="og:title" content="Work_documented"/><meta property="og:description" content="This is file contains all the steps done for the master thesis"/><meta property="og:url" content="https://vishnushiri02.github.io/SARS-Cov2/notes/r423m96u71ix4pb458fk8u2/"/><meta property="og:type" content="article"/><meta property="article:published_time" content="11/17/2023"/><meta property="article:modified_time" content="3/20/2024"/><link rel="canonical" href="https://vishnushiri02.github.io/SARS-Cov2/notes/r423m96u71ix4pb458fk8u2/"/><meta name="next-head-count" content="14"/><link rel="preload" href="/SARS-Cov2/_next/static/css/b185cac82247bc6c.css" as="style"/><link rel="stylesheet" href="/SARS-Cov2/_next/static/css/b185cac82247bc6c.css" data-n-g=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/SARS-Cov2/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js"></script><script src="/SARS-Cov2/_next/static/chunks/webpack-1cb654aa556e0993.js" defer=""></script><script src="/SARS-Cov2/_next/static/chunks/framework-28c999baf2863c3d.js" defer=""></script><script src="/SARS-Cov2/_next/static/chunks/main-fa154d5c5ddc4bf7.js" defer=""></script><script src="/SARS-Cov2/_next/static/chunks/pages/_app-0218995cbb01ab10.js" defer=""></script><script src="/SARS-Cov2/_next/static/chunks/935-4dee79e80b8641c6.js" defer=""></script><script src="/SARS-Cov2/_next/static/chunks/6-50972def09142ee2.js" defer=""></script><script src="/SARS-Cov2/_next/static/chunks/pages/notes/%5Bid%5D-78d472fa3b924116.js" defer=""></script><script src="/SARS-Cov2/_next/static/npK0PGR6Fa7mwLej5tvS3/_buildManifest.js" defer=""></script><script src="/SARS-Cov2/_next/static/npK0PGR6Fa7mwLej5tvS3/_ssgManifest.js" defer=""></script></head><body><div id="__next" data-reactroot=""><section class="ant-layout" style="width:100%;min-height:100%"><header class="ant-layout-header" style="position:fixed;isolation:isolate;z-index:1;width:100%;border-bottom:1px solid #d4dadf;height:64px;padding:0 24px 0 2px"><div class="ant-row ant-row-center" style="max-width:992px;justify-content:space-between;margin:0 auto"><div style="display:flex" class="ant-col ant-col-xs-20 ant-col-sm-4"></div><div class="ant-col gutter-row ant-col-xs-0 ant-col-sm-20 ant-col-md-20 ant-col-lg-19"><div class="ant-select ant-select-lg ant-select-auto-complete ant-select-single ant-select-allow-clear ant-select-show-search" style="width:100%"><div class="ant-select-selector"><span class="ant-select-selection-search"><input type="search" autoComplete="off" class="ant-select-selection-search-input" role="combobox" aria-haspopup="listbox" aria-owns="undefined_list" aria-autocomplete="list" aria-controls="undefined_list" aria-activedescendant="undefined_list_0" value=""/></span><span class="ant-select-selection-placeholder">For full text search please use the &#x27;?&#x27; prefix. e.g. ? Onboarding</span></div></div></div><div style="display:none;align-items:center;justify-content:center" class="ant-col ant-col-xs-4 ant-col-sm-4 ant-col-md-0 ant-col-lg-0"><span role="img" aria-label="menu" style="font-size:24px" tabindex="-1" class="anticon anticon-menu"><svg viewBox="64 64 896 896" focusable="false" data-icon="menu" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"></path></svg></span></div></div></header><section class="ant-layout" style="margin-top:64px;display:flex;flex-direction:row"><div class="site-layout-sidebar" style="flex:0 0 auto;width:calc(max((100% - 992px) / 2, 0px) + 200px);min-width:200px;padding-left:calc((100% - 992px) / 2)"><aside class="ant-layout-sider ant-layout-sider-dark" style="position:fixed;overflow:auto;height:calc(100vh - 64px);background-color:transparent;flex:0 0 200px;max-width:200px;min-width:200px;width:200px"><div class="ant-layout-sider-children"></div></aside></div><main class="ant-layout-content side-layout-main" style="max-width:1200px;min-width:0;display:block"><div style="padding:0 24px"><div class="main-content" role="main"><div class="ant-row"><div class="ant-col ant-col-24"><div class="ant-row" style="margin-left:-10px;margin-right:-10px"><div style="padding-left:10px;padding-right:10px" class="ant-col ant-col-xs-24 ant-col-md-18"><div><h1 id="work_documented">Work_documented<a aria-hidden="true" class="anchor-heading icon-link" href="#work_documented"></a></h1>
<h1 id="objective">Objective<a aria-hidden="true" class="anchor-heading icon-link" href="#objective"></a></h1>
<p>Is it possible to predict the positions that are vulnerable for forth coming mutations.</p>
<h2 id="steps-in-the-workflow">Steps in the workflow<a aria-hidden="true" class="anchor-heading icon-link" href="#steps-in-the-workflow"></a></h2>
<div class="mermaid">
  graph TD;
1[start]--Based on data quantity-->A[Choosing countries for the study]--checking for extreme low value outliers-->B[Downloading sequence and meta data from these countries for 22 months]--Mapping lineages to parental lineage-->C[Finding the difference in the trend of variant frequencies in these countries]
</div>
<ul>
<li><a href="/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#1-whyhow-was-this-particular-time-period-jan22-oct23-chosen-while-pandemic-has-been-prevelant-2020-onwards">Question on date</a></li>
</ul>
<h2 id="data-acquisition">Data acquisition<a aria-hidden="true" class="anchor-heading icon-link" href="#data-acquisition"></a></h2>
<ul>
<li>List of locations available in GISAID was obtained from GISAIDR Git repo - <a href="https://github.com/Wytamma/GISAIDR/blob/master/GISAID_LOCATIONS.txt">GISAID_LOCATIONS.txt</a></li>
<li>The following points/steps are coded in the file <code>Work/Data_Analysis/Monthwise_data_submission.Rmd</code>
<ul>
<li>From the GISAID, the total number of <strong><em>high</em></strong>, <strong><em>coverge</em></strong> completed sequence deposited by each country (200+) in the past 22 months(Jan2022-Oct2023) were retrieved in a month wise fasion using the R Package GISAIDR. The table is in <code>Work/Data_Analysis/country_monthly_submission.csv</code></li>
<li>Countries that had more than 10,000 sequences deposited in total in the past 22 months were alone considered further.</li>
<li>This included 10 countries : India, South Korea, Denmark, Germany, Norway, Spain, United Kingdom, Canada, USA, Australia. <code>Work/Data_Analysis/chosen_ten_country_submission.csv</code></li>
<li>The trend of the time series submission statistics of these countries were plotted. It is present in the file <code>Work/Data_Analysis/plots/ten_Country_data_trend_plots.pdf</code>
<img src="/SARS-Cov2/assets/plots/Country_trend_plots.png" alt="Trends in 10 countries"></li>
<li>Outlier detection and pruning of the observation horizon(Time period considered): Extreme low values were searched for, using the <a href="/SARS-Cov2/notes/v4t96j5kch5mq12272qsyih#hampler-filter">Hampler Filter</a> (one of the <a href="/SARS-Cov2/notes/v4t96j5kch5mq12272qsyih#outlier-estimation-and-methods">Outlier Estimation and methods</a>). outliers_mad from Routliers package was used . Since there were no extreme low values found in any of the countries, the observation time period was not pruned.</li>
<li>The sequence data for 10 countries were downloaded. There are files with crude data - all data retrieved from GISAID for each of the accesssion ID (29 cols of data) and there are files with essential columns: strain,virus,collection_date,country,pangolineage,sequence, length of the sequence, GISAID accession_id. The crude data for 10 countries are present 10 files in the folder <code>Work/Data_Analysis/ten_country_crude_data_seq</code> and the essential column files are in the folder <code>Work/Data_Analysis/ten_country_ess_data</code></li>
<li>The Rmd also has a codeblock for downloading data with AA_substitution. This data was needed for mapping lineages to their parental lineage. For this I used the modified download script(modified by me) <code>Work/Data_Analysis/Modified_GISAIDRdownload.R</code>.
<blockquote>
<p>This can be done by <code>fix(Download)</code> and modifying the script as we need</p>
</blockquote>
</li>
<li>This mutation data are stored in <code>Work/Data_Analysis/ten_country_mut_data</code></li>
</ul>
</li>
</ul>
<blockquote>
<p>There were some inconsistencies in the Date entries (only year no month or year and month an no date) such entries were removed.
On the course of doing this it was identified that the number of entries that was downloaded for each countries do not exactly match the numbers in <code>Work/Data_Analysis/chosen_ten_country_submission.csv</code> excluding few countries.
After removing the inconsistent date entries the numbers(downloaded and number in file) for India match.
The inconsistencies are reported in the file <code>Work/Data_Analysis/inconsistencies_in_data.numbers</code>.
The number of downloaded entries are more than then numbers in file.
After looking into it, it was observed that there were few more entries added to these countries in the time gap(around a week) between the compiling <code>Work/Data_Analysis/chosen_ten_country_submission.csv</code> and downloading data. <a href="/SARS-Cov2/notes/o7n9flqgbqf3xbm437bc0ih">checking inconsistencies</a></p>
<h2 id="summary">Summary<a aria-hidden="true" class="anchor-heading icon-link" href="#summary"></a></h2>
<p>Summary of the downloaded data</p>
</blockquote>
<div class="table-responsive">

















































































<table><thead><tr><th><strong>Country</strong></th><th><strong>Total number of entries downloaded (Jan22-Oct23)</strong></th><th><strong>Entries after removing the date inconsistencies</strong></th><th><strong>Number of unassigned</strong></th><th><strong>Percentage of unassigned</strong></th></tr></thead><tbody><tr><td><strong>Asia / India</strong></td><td>18106</td><td>15783</td><td>0</td><td>0 %</td></tr><tr><td><strong>Asia / South Korea</strong></td><td>13808</td><td>13808</td><td>0</td><td>0 %</td></tr><tr><td><strong>Europe / Denmark</strong></td><td>163900</td><td>163900</td><td>7</td><td>0,004 %</td></tr><tr><td><strong>Europe / Germany</strong></td><td>66374</td><td>66236</td><td>0</td><td>0 %</td></tr><tr><td><strong>Europe / Norway</strong></td><td>12264</td><td>12264</td><td>0</td><td>0 %</td></tr><tr><td><strong>Europe / Spain</strong></td><td>17991</td><td>17921</td><td>0</td><td>0 %</td></tr><tr><td><strong>Europe / United Kingdom</strong></td><td>32835</td><td>30597</td><td>74</td><td>0,242 %</td></tr><tr><td><strong>North America / Canada</strong></td><td>20791</td><td>20791</td><td>0</td><td>0 %</td></tr><tr><td><strong>North America / USA</strong></td><td>75487</td><td>74971</td><td>5</td><td>0,007 %</td></tr><tr><td><strong>Oceania / Australia</strong></td><td>13267</td><td>13257</td><td>5</td><td>0,038 %</td></tr></tbody></table></div>
<p><a href="/SARS-Cov2/notes/r423m96u71ix4pb458fk8u2#dealing-with-the-unassigned">Note on the unassigned</a></p>
<h2 id="mapping-lineages">Mapping lineages<a aria-hidden="true" class="anchor-heading icon-link" href="#mapping-lineages"></a></h2>
<ul>
<li>
<p>Mapping of lineages to their parental lineage is done based on the Pangolin lineage naming string and the spike mutations:</p>
<ul>
<li>From the 10 files of <code>Work/Data_Analysis/ten_country_mut_data</code> the columns corresponding to the mutations and pango lineage were extracted and stored in a different file <code>Work/Data_Analysis/ten_country_mut_data/ten_country_lineage_mut.csv</code>.
This is done using awk command</li>
</ul>
</li>
</ul>
<pre class="language-bash"><code class="language-Bash"><span class="token function">awk</span> -F<span class="token string">";"</span> <span class="token string">'{gsub(/[()]/,"",$6);print $5";"$6}'</span> 
ten_country_mut_data/* <span class="token punctuation">\</span>
<span class="token operator">></span>ten_country_mut_data/ten_country_lineage_mut.csv
</code></pre>
<blockquote>
<p>gsub is used to remove the brackets in the AA_mutations column.<br>
Columns of the CSV files are separated by a semicolon ';'</p>
</blockquote>
<ul>
<li>
<p>All the following steps are coded in the Rmd file <code>Work/Data_Analysis/Lineage_mapping.Rmd</code></p>
<ul>
<li>
<p>All the mutations pertaining to each of the lineage were combined.</p>
</li>
<li>
<p>From this only the spike mutations were considered for the downstream work.</p>
</li>
<li>
<p>Jaccard index (intersections of sets/Union of sets). If the Jaccard value calculated  is less than 0.5 then the lineages being compared are either considered as parental or neighbour depending on their pangolin string. <a href="/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#3-the-jaccard-index-treshold-chosen-to-decide-a-parent-or-a-neighbour-is-50-is-this-okay">Question on the treshold</a></p>
<ul>
<li>The Lineages were mapped to parental lineage based on the the calculated jaccard value which is based on the number of common spike mutations between pangolin lineages that have same names dropping the last character in the sublineage. VOC/VUM/VOI were explicitely mapped based on the list taken from GISAID.</li>
</ul>
</li>
<li>
<p>For Pangolin lineages that are considered as VOI/VUM/VOC, the  mapping is directly given in GISAID. This can be found in the file <code>Work/Data_Analysis/GISAID_VOI_VOC_VOM_list.txt</code>.</p>
<ul>
<li>
<p>This list was obtained by first downloading the Clade/Lineage,variants(tsv) from the GISAID Downloads prompt.</p>
</li>
<li>
<p>This tsv file has multiple columns. Interesting columns were type and Value. Using the following bash commands the list of GISAID_VOI_VOC_VOMlist.txt was compiled</p>
<pre class="language-bash"><code class="language-bash"><span class="token function">awk</span> -F<span class="token punctuation">\</span>t <span class="token string">'{if($3=="Variant") print $4}'</span> <span class="token punctuation">\</span>
gisaid_variants_statistics_from_gisaid.tsv <span class="token operator">|</span> <span class="token punctuation">\</span>
<span class="token function">sort</span> -u <span class="token operator">></span>GISAID_VOI_VOC_VOM_list.txt
</code></pre>
</li>
<li>
<p>The resulting text has long lines like <em><strong>VOI GRA (EG.5+EG.5.*) first detected in Indonesia/France</strong></em>
from this only the variant and sublineage names EG.5+EG.5.*  are retained while other texts including the brackets are removed(manually).</p>
</li>
<li>
<p>Lines are rearranged in such a way that parental variant comes after the subvariant, example: XBB+XBB.*    would come only after XBB.1.16+XBB.1.16.*. This rearrangement was done manually.
<img src="/SARS-Cov2/assets/Pics/Mapping_lineage_idea.png" alt="mapping lineages idea"></p>
</li>
</ul>
</li>
<li>
<p>In total there are 1628 unique lingeages which includes variants also. Running the mapping algorithm on this gives 715 lineages mapped to 16 variants and 913 lineages getting mapped to 531 lineages which includes Unknown.</p>
</li>
</ul>
<blockquote>
<p>If muation list is obtained using the outbreakinfo function getMutationsByLineage then the results vary for the non-variant lineages. 912(without unknown) lineages gets mapped to 676 lineages. This is present in assigning_the_unassigned.RMD</p>
</blockquote>
</li>
</ul>
<h2 id="frequency-computation">Frequency computation<a aria-hidden="true" class="anchor-heading icon-link" href="#frequency-computation"></a></h2>
<ul>
<li>After the lineage mapping, for each country the frequency of presence of each parental_lineage(which also includes the sublienages) in a month was calculated (<span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mfrac><mrow><mi>c</mi><mi>o</mi><mi>u</mi><mi>n</mi><mi>t</mi><mtext> </mtext><mi>o</mi><mi>f</mi><mtext> </mtext><mi>l</mi><mi>i</mi><mi>n</mi><mi>e</mi><mi>a</mi><mi>g</mi><mi>e</mi><mtext> </mtext><mi>B</mi><mtext> </mtext><mi>i</mi><mi>n</mi><mtext> </mtext><mi>j</mi><mi>u</mi><mi>l</mi></mrow><mrow><mi>t</mi><mi>o</mi><mi>t</mi><mi>a</mi><mi>l</mi><mtext> </mtext><mi>e</mi><mi>n</mi><mi>t</mi><mi>r</mi><mi>i</mi><mi>e</mi><mi>s</mi><mtext> </mtext><mi>i</mi><mi>n</mi><mtext> </mtext><mi>j</mi><mi>u</mi><mi>l</mi></mrow></mfrac></mrow><annotation encoding="application/x-tex">\frac{count\space of\space lineage\space B\space in\space jul}{total\space entries\space in\space jul}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.4133em;vertical-align:-0.4811em;"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.9322em;"><span style="top:-2.655em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">t</span><span class="mord mathnormal mtight">o</span><span class="mord mathnormal mtight">t</span><span class="mord mathnormal mtight">a</span><span class="mord mathnormal mtight" style="margin-right:0.01968em;">l</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">e</span><span class="mord mathnormal mtight">n</span><span class="mord mathnormal mtight">t</span><span class="mord mathnormal mtight" style="margin-right:0.02778em;">r</span><span class="mord mathnormal mtight">i</span><span class="mord mathnormal mtight">es</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">in</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight" style="margin-right:0.01968em;">l</span></span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.4461em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">co</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">n</span><span class="mord mathnormal mtight">t</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">o</span><span class="mord mathnormal mtight" style="margin-right:0.10764em;">f</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight" style="margin-right:0.01968em;">l</span><span class="mord mathnormal mtight">in</span><span class="mord mathnormal mtight">e</span><span class="mord mathnormal mtight">a</span><span class="mord mathnormal mtight" style="margin-right:0.03588em;">g</span><span class="mord mathnormal mtight">e</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight" style="margin-right:0.05017em;">B</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">in</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight" style="margin-right:0.05724em;">j</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight" style="margin-right:0.01968em;">l</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.4811em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span>) with confidence interval and plotted.MultinomCI from DescTools package is used to compute the simultaneous <a href="/SARS-Cov2/notes/v4t96j5kch5mq12272qsyih#confidence-interval">Confidence interval</a>. This function also calculates frequency/proportion so a seperate computation was not required.95% confidence interval with method sisonglaz was computed. The plots are stored in PDF file, 1 plot for each country so 10 pages - <code>Work/Data_Analysis/plots/country_wise_frequency_trend.pdf</code> without CI and <code>Work/Data_Analysis/plots/country_wise_frequency_trend_CI.pdf</code> with CI.</li>
<li>To make it more convenient plots for each Parental_lineage/variant grouped by country was plotted to do the analysis. These are present in the pdf files - <code>Work/Data_Analysis/plots/Variant_wise_trend_all_countries.pdf</code> without CI and <code>Work/Data_Analysis/plots/Variant_wise_trend_all_countries_CI.pdf</code> with CI</li>
<li>These plots are analysed to bring out the difference in the variant trend among countries. An entire spreedsheet was developed manually basically describing the variant trends in words <code>Work/Data_Analysis/plot_Analysis_report.numbers</code>, and the summary of the analysis is in <a href="/SARS-Cov2/notes/1hnn8maoednnxdw5feqd4kq">Analysis of the varient trend plots</a> file.</li>
<li>To not miss on the difference of pandemic in countries the lineages which are not variants were also used for comparison like the variants. There are 300+ plots in the file each page corresponding to one lineage which is not a variants and the data is grouped by country. This is presetn in <code>Work/Data_Analysis/plots/Lineages_no_var_freq_countrywise_CI</code> with CI and in <code>Work/Data_Analysis/plots/Lineages_no_var_freq_countrywise</code> without CI.</li>
<li>From these plots trends that look interesting are chosen and noted in <a href="/SARS-Cov2/notes/1hnn8maoednnxdw5feqd4kq">Analysis of the varient trend plots</a></li>
</ul>
<h2 id="dealing-with-the-unassigned">Dealing with the unassigned<a aria-hidden="true" class="anchor-heading icon-link" href="#dealing-with-the-unassigned"></a></h2>
<ul>
<li>For few entries of the GISAID downloaded data the lineage was not assigned. The lineage column had either a blank space or it was termed as unassigned. There were in total 91 unassigned entries in the downloaded 10 country data.</li>
<li>Denmark 7 entries ,UK 74 entries, USA 5 entries, Australia 5 which is Denmark 0.004%, UK 0.242%, USA 0.007%, Australia 0.038% of the entries in the respective countries.</li>
<li>Majority of the unassigned entries were collected in Oct23</li>
<li>To get the possible assignment for these entries few methods were tried out.</li>
<li>First among them was to calculate distance between the unassigned and all other lineages based on their spike mutations. For this process the characteristics mutation for each lineage was obtained using the outbreakinfo function <em><strong>getMutationsByLineage</strong></em>. Lineages that had less distance to each of the unassigned entries were considered as possible assignment.</li>
<li>Though this method seems to work when the method was checked with entries with known assignments, the method did not perform well.</li>
<li>The main reason that this wouldn't work is beacuse not all the characteristic spike mutation is always present in the GISAID downloaded data, so when distance is computed there is always discrepencies.</li>
<li>Hence to go with an accurate method genome sequences of these unassigned entries were obtained and a fasta file was compiled.</li>
<li>This file <code>unassigned_sequences.fasta</code> was uploaded in the <a href="https://pangolin.cog-uk.io">https://pangolin.cog-uk.io</a> to get the assignment.</li>
<li>Once the assignment was obtained it was added to the dataset. This part is included in the fill_parental function of lineage_mapping. The trends were reanalysed after assigning the unassigned.
<blockquote>
<p>The <a href="https://pangolin.cog-uk.io">https://pangolin.cog-uk.io</a> was not able to analyse the 5 sequences from australia. No reasons was provided.</p>
</blockquote>
</li>
</ul>
<p><img src="/SARS-Cov2/assets/Pics/pangolin_assignment_analysis.png" alt="pangolin_assignment"></p>
<p><img src="/SARS-Cov2/assets/Pics/pangolin_assignment.png" alt="pangolin_assignemt_details"></p>
<ul>
<li>All these are present in the <code>Work/Data_Analysis/assigning_the_unassigned.Rmd</code></li>
</ul>
<h2 id="finding-positions-under-pressure-big-goal">Finding positions under pressure (BIG GOAL)<a aria-hidden="true" class="anchor-heading icon-link" href="#finding-positions-under-pressure-big-goal"></a></h2>
<p>The big goal is to find the positions under pressure. To obtain this, firstly the frequency of  mutation on each position of Spike RBD and NTD Epitopes are calculated. (RBD spike mutations in position 330-530, NTD(14-20,140-158,245-264) mutations)(<a href="/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#5-aaccording-to-uniprot-the-rbd-region-in-spike--is-319-541aa">Question on the position</a>) For this the aa_substitution data obtained from GISAID was used.
<img src="/SARS-Cov2/assets/Pics/Frequency_interpolation.png" alt="frequency interpolation"></p>
<p><strong>STEPS:</strong></p>
<ul>
<li>For Each country the mutation data from GISAID has been used as the input.</li>
<li>This data consist of the lineage, collected date, location of collection, a string of aa_substitution.</li>
<li>All the aa_substitution strings that are collected on the same day are considered together. This also gives the number of entries collected in a day = number sequences colleceted on a day.</li>
<li>From these strings the spike RBD, NTD epitope mutations are alone extracted using getintrmut function.</li>
<li>The data frame gets reduced to date,number of sequences,spike RBD/NTD mutations string.</li>
<li>To calculate the frequency of a mutation for a time step, all the possible unique spike RBD, NTD mutations present in the country is first compiled.</li>
<li>Then the frequency of each of this mutation for a day is calculated.
<blockquote>
<p>Frequency of pos_373 on 01-01-2022 = <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mfrac><mrow><mi>c</mi><mi>o</mi><mi>u</mi><mi>n</mi><mi>t</mi><mtext> </mtext><mi>o</mi><mi>f</mi><mtext> </mtext><mi>m</mi><mi>u</mi><mi>t</mi><mi>a</mi><mi>t</mi><mi>i</mi><mi>o</mi><mi>n</mi><mi>s</mi><mtext> </mtext><mi>i</mi><mi>n</mi><mtext> </mtext><mi>p</mi><mi>o</mi><mi>s</mi><mi mathvariant="normal">_</mi><mn>373</mn><mtext> </mtext><mi>o</mi><mi>n</mi><mtext> </mtext><mn>01</mn><mo>−</mo><mn>01</mn><mo>−</mo><mn>2022</mn></mrow><mrow><mi>N</mi><mi>u</mi><mi>m</mi><mi>b</mi><mi>e</mi><mi>r</mi><mtext> </mtext><mi>o</mi><mi>f</mi><mtext> </mtext><mi>s</mi><mi>e</mi><mi>q</mi><mi>u</mi><mi>e</mi><mi>n</mi><mi>c</mi><mi>e</mi><mi>s</mi><mtext> </mtext><mi>o</mi><mi>n</mi><mtext> </mtext><mn>01</mn><mo>−</mo><mn>01</mn><mo>−</mo><mn>2022</mn></mrow></mfrac></mrow><annotation encoding="application/x-tex">\frac{count\space of\space mutations\space in\space pos\_373\space on\space 01-01-2022}{Number\space of\space sequences\space on\space 01-01-2022}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.4942em;vertical-align:-0.4811em;"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.0131em;"><span style="top:-2.655em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.10903em;">N</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">mb</span><span class="mord mathnormal mtight" style="margin-right:0.02778em;">er</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">o</span><span class="mord mathnormal mtight" style="margin-right:0.10764em;">f</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">se</span><span class="mord mathnormal mtight" style="margin-right:0.03588em;">q</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">e</span><span class="mord mathnormal mtight">n</span><span class="mord mathnormal mtight">ces</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">o</span><span class="mord mathnormal mtight">n</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mtight">01</span><span class="mbin mtight">−</span><span class="mord mtight">01</span><span class="mbin mtight">−</span><span class="mord mtight">2022</span></span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.527em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">co</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">n</span><span class="mord mathnormal mtight">t</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">o</span><span class="mord mathnormal mtight" style="margin-right:0.10764em;">f</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">m</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">t</span><span class="mord mathnormal mtight">a</span><span class="mord mathnormal mtight">t</span><span class="mord mathnormal mtight">i</span><span class="mord mathnormal mtight">o</span><span class="mord mathnormal mtight">n</span><span class="mord mathnormal mtight">s</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">in</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">p</span><span class="mord mathnormal mtight">os</span><span class="mord mtight">_373</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mathnormal mtight">o</span><span class="mord mathnormal mtight">n</span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mord mtight">01</span><span class="mbin mtight">−</span><span class="mord mtight">01</span><span class="mbin mtight">−</span><span class="mord mtight">2022</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.4811em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></p>
</blockquote>
</li>
<li>To have the frequency data for everyday from Jan 1 2022 - Oct 31 2023, linear <a href="/SARS-Cov2/notes/v4t96j5kch5mq12272qsyih#interpolation">interpolation</a> is done using the approximate method. The result of the interpolation is added to the corresponding date in the country data_frame. This step is necessary to fill in the data gaps.</li>
</ul>
<p><a href="/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#4-why-do-we-do-linear-interpolation-why-not-spline-interpolation">Question on interpolation</a></p>
<h2 id="computing-the-pressure-on-the-position">Computing the pressure on the position<a aria-hidden="true" class="anchor-heading icon-link" href="#computing-the-pressure-on-the-position"></a></h2>
<p>With the formula (given by Prof.Max) the pressure on each position from t<sub>0</sub> to t was computed
<span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mspace linebreak="newline"></mspace><mi>P</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>t</mi><mo stretchy="false">)</mo><mo>=</mo><msubsup><mo>∑</mo><mrow><mi>s</mi><mo>=</mo><msub><mi>t</mi><mn>0</mn></msub></mrow><mi>t</mi></msubsup><msup><mrow><mi>exp</mi><mo>⁡</mo></mrow><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>t</mi><mo>−</mo><mi>s</mi><mo stretchy="false">]</mo></mrow></msup><mo>×</mo><mi>f</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>s</mi><mo stretchy="false">)</mo><mspace linebreak="newline"></mspace></mrow><annotation encoding="application/x-tex">\\ P(pos,t)=\sum_{s=t_0}^{t}\exp^{-k[t-s]}\times f(pos,s) \\</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="mspace newline"></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">t</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:1.3333em;vertical-align:-0.3998em;"></span><span class="mop"><span class="mop op-symbol small-op" style="position:relative;top:0em;">∑</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.9335em;"><span style="top:-2.4003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">s</span><span class="mrel mtight">=</span><span class="mord mtight"><span class="mord mathnormal mtight">t</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3173em;"><span style="top:-2.357em;margin-left:0em;margin-right:0.0714em;"><span class="pstrut" style="height:2.5em;"></span><span class="sizing reset-size3 size1 mtight"><span class="mord mtight">0</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.143em;"><span></span></span></span></span></span></span></span></span></span><span style="top:-3.2029em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.3998em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mop"><span class="mop">exp</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.888em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">t</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">s</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord">×</span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">s</span><span class="mclose">)</span></span><span class="mspace newline"></span></span></span></span></p>
<ul>
<li>Where the f(pos,t) is the frequency of the position on time t.</li>
<li><span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>e</mi><mi>x</mi><msup><mi>p</mi><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>t</mi><mo>−</mo><mi>s</mi><mo stretchy="false">]</mo></mrow></msup></mrow><annotation encoding="application/x-tex">exp^{-k[t-s]}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.0824em;vertical-align:-0.1944em;"></span><span class="mord mathnormal">e</span><span class="mord mathnormal">x</span><span class="mord"><span class="mord mathnormal">p</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.888em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">t</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">s</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span></span></span></span></span> is the discount factor - mutation frequencies that occurred [t-s] days ago get discounted by the half life of neutralising antibodies.</li>
<li><span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>∼</mo><mfrac><mrow><mi>l</mi><mi>n</mi><mo stretchy="false">(</mo><mn>2</mn><mo stretchy="false">)</mo></mrow><mrow><mn>45</mn><mo>+</mo><mn>14</mn></mrow></mfrac></mrow><annotation encoding="application/x-tex">k\sim \frac{ln(2)}{45+14}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6944em;"></span><span class="mord mathnormal" style="margin-right:0.03148em;">k</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">∼</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:1.4133em;vertical-align:-0.4033em;"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.01em;"><span style="top:-2.655em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">45</span><span class="mbin mtight">+</span><span class="mord mtight">14</span></span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.485em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.01968em;">l</span><span class="mord mathnormal mtight">n</span><span class="mopen mtight">(</span><span class="mord mtight">2</span><span class="mclose mtight">)</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.4033em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></li>
<li>By suggestion of the Prof. the vector for discount factor was first computed. For this the date range for each of the country_df was found. If the difference between the start day and the end day is 9 then [t-s] could be in the range 0-9. Hence with this as base the discount factor was computed for [t-s] ranging 0-[difference between the start day to end day in the dataframe]. All these values are stored in a vector.</li>
<li>From the start date to the end date the pressure was calculated for everday in the following manner</li>
<li>
<div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mspace linebreak="newline"></mspace><mi>P</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>1</mn><mo stretchy="false">)</mo><mo>=</mo><msup><mrow><mi>exp</mi><mo>⁡</mo></mrow><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>d</mi><mn>1</mn><mo>−</mo><mi>d</mi><mn>1</mn><mo stretchy="false">]</mo></mrow></msup><mo>×</mo><mi>f</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>1</mn><mo stretchy="false">)</mo><mspace linebreak="newline"></mspace><mi>P</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>2</mn><mo stretchy="false">)</mo><mo>=</mo><msup><mrow><mi>exp</mi><mo>⁡</mo></mrow><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>d</mi><mn>2</mn><mo>−</mo><mi>d</mi><mn>1</mn><mo stretchy="false">]</mo></mrow></msup><mo>×</mo><mi>f</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>1</mn><mo stretchy="false">)</mo><mo>+</mo><msup><mrow><mi>exp</mi><mo>⁡</mo></mrow><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>d</mi><mn>2</mn><mo>−</mo><mi>d</mi><mn>2</mn><mo stretchy="false">]</mo></mrow></msup><mo>×</mo><mi>f</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>2</mn><mo stretchy="false">)</mo><mspace linebreak="newline"></mspace><mi>P</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>3</mn><mo stretchy="false">)</mo><mo>=</mo><msup><mrow><mi>exp</mi><mo>⁡</mo></mrow><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>d</mi><mn>3</mn><mo>−</mo><mi>d</mi><mn>1</mn><mo stretchy="false">]</mo></mrow></msup><mo>×</mo><mi>f</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>1</mn><mo stretchy="false">)</mo><mo>+</mo><msup><mrow><mi>exp</mi><mo>⁡</mo></mrow><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>d</mi><mn>3</mn><mo>−</mo><mi>d</mi><mn>2</mn><mo stretchy="false">]</mo></mrow></msup><mo>×</mo><mi>f</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>2</mn><mo stretchy="false">)</mo><mo>+</mo><msup><mrow><mi>exp</mi><mo>⁡</mo></mrow><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>d</mi><mn>3</mn><mo>−</mo><mi>d</mi><mn>3</mn><mo stretchy="false">]</mo></mrow></msup><mo>×</mo><mi>f</mi><mo stretchy="false">(</mo><mi>p</mi><mi>o</mi><mi>s</mi><mo separator="true">,</mo><mi>d</mi><mn>3</mn><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">\\ P(pos,d1)=\exp^{-k[d1-d1]}\times f(pos,d1) \\ P(pos,d2)=\exp^{-k[d2-d1]}\times f(pos,d1) + \exp^{-k[d2-d2]}\times f(pos,d2)\\ P(pos,d3)=\exp^{-k[d3-d1]}\times f(pos,d1) + \exp^{-k[d3-d2]}\times f(pos,d2) + \exp^{-k[d3-d3]}\times f(pos,d3)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="mspace newline"></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">1</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:1.188em;vertical-align:-0.25em;"></span><span class="mop"><span class="mop">exp</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.938em;"><span style="top:-3.113em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">1</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">1</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord">×</span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">1</span><span class="mclose">)</span></span><span class="mspace newline"></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">2</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:1.188em;vertical-align:-0.25em;"></span><span class="mop"><span class="mop">exp</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.938em;"><span style="top:-3.113em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">2</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">1</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord">×</span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">1</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222em;"></span></span><span class="base"><span class="strut" style="height:1.188em;vertical-align:-0.25em;"></span><span class="mop"><span class="mop">exp</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.938em;"><span style="top:-3.113em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">2</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">2</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord">×</span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">2</span><span class="mclose">)</span></span><span class="mspace newline"></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">3</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:1.188em;vertical-align:-0.25em;"></span><span class="mop"><span class="mop">exp</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.938em;"><span style="top:-3.113em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">3</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">1</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord">×</span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">1</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222em;"></span></span><span class="base"><span class="strut" style="height:1.188em;vertical-align:-0.25em;"></span><span class="mop"><span class="mop">exp</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.938em;"><span style="top:-3.113em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">3</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">2</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord">×</span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">2</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222em;"></span></span><span class="base"><span class="strut" style="height:1.188em;vertical-align:-0.25em;"></span><span class="mop"><span class="mop">exp</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.938em;"><span style="top:-3.113em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">3</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">d</span><span class="mord mtight">3</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord">×</span><span class="mord mathnormal" style="margin-right:0.10764em;">f</span><span class="mopen">(</span><span class="mord mathnormal">p</span><span class="mord mathnormal">os</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">d</span><span class="mord">3</span><span class="mclose">)</span></span></span></span></span></div>
</li>
<li>This was done for all the RBD, NTD positions in a country and repeated for all the 10 countries.</li>
<li>The output will have one column with the RBD and NTD positions and with columns as many as the number of days with data in the country
<img src="/SARS-Cov2/assets/Pics/Pressure_computed_india.png" alt="Pressure output"></li>
</ul>
<h2 id="masking">Masking<a aria-hidden="true" class="anchor-heading icon-link" href="#masking"></a></h2>
<ul>
<li>The objective of masking to find the exposed positions among the RBD, NTD spike positions.</li>
<li>To know the exposed residues the solvent accessibility of each of the residues in the spike protein were found. <a href="/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#6-if-a-rbd-spike-position-in-the-wildtype-is-occupied-by-a-hydrophobic-residue-and-it-is-replaced-by-hydrophilic-residue-the-solvent-accessibility-might-change-probably-due-to-the-difference-in-the-fold--in-that-case-should-we-study-these-positions-in-each-of-the-voi">Question on solvent acessibility</a></li>
<li>To find solvent accessibility of the protein various tools were utilitsed. This can be found here<a href="/SARS-Cov2/notes/g9181ia6dhnmpqfklf2mwit">Finding_surface_residues</a></li>
<li>From the output of each of the tool Spike RBD,NTD surface positions were found. This process was direct in the case of the output from GetArea and Netsurf3.0 but in the case of DSSP, relative solvent accessibility was computed from the absolute solvent accessibility in the dssp output file. Using this computed relative solvent accessibility the surface residues were found.</li>
</ul>
<blockquote>
<p>The categorising treshold is set as 25% similar to Netsurf3.0. This would need advice.</p>
</blockquote>
<ul>
<li>
<p>The number of surface residues flagged by the 3 tools vary.</p>
<ul>
<li>Dssp : 74 residues, NetSur3.0 : 123 residues, GetArea : 50 residues. All three tools flags 36 positions in common, GetArea and netsurf flags 38 common positions, dssp-getarea 48 positions in common, dssp-netsurf 54 positions in common.</li>
<li>Dssp : 12 NTD, 62 RBD</li>
<li>GetArea: 8 NTD, 42 RBD</li>
<li>Netsurf: 22 NTD, 101 RBD</li>
</ul>
</li>
<li>
<p>Residues flagged by each of the tools are fished from the ten country's position under pressure dataframe. The numbers are tabulated .<img src="/SARS-Cov2/assets/Pics/methods_comparison.png" alt="Methods comparison"></p>
</li>
</ul>
<p>The positions being picked by each of the tools are compared with the positions given by Prof.Max(screenshot). The tabulation is available in <code>Work/Data_Analysis/Positions_being_picked.numbers</code></p>
<h2 id="visualization-of-the-positions-under-pressure">Visualization of the positions under pressure<a aria-hidden="true" class="anchor-heading icon-link" href="#visualization-of-the-positions-under-pressure"></a></h2>
<p>Like previous variant trend visualization, the pressure trend was also visualised using geom_line in two ways, positionwise and countrywise. These plots are present in <code>Work/Data_Analysis/pressure_plots</code>. There are 6 plots - 3 countrywise, 3 positionsise. The 3 denotes the 3 masking tool outputs that were used to generate the plots.</p>
<h2 id="entropy-for-each-positions">Entropy for each positions:<a aria-hidden="true" class="anchor-heading icon-link" href="#entropy-for-each-positions"></a></h2>
<p>The amino acid substitution on a position is not constant. Example: on position 14 mutations found in india include Spike_Q14H,Spike_Q14del,Spike_Q14R. So calculating the number of such prevailing mutations on a position is the entropy. This was calculated to weigh down the pressure on a postiion .... Not processed further yet</p>
<h2 id="neutralisation-probability">Neutralisation probability:<a aria-hidden="true" class="anchor-heading icon-link" href="#neutralisation-probability"></a></h2>
<p>For knowing how impactful would a mutation on a site be, the probaility of neutralisation is being calculated. The Probability of neutralisation of a variant y by the antibodies elicited by variant x at time t.</p>
<div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mspace linebreak="newline"></mspace><msub><mi>P</mi><mrow><mi>N</mi><mi>e</mi><mi>u</mi><mi>t</mi></mrow></msub><mo stretchy="false">(</mo><mi>t</mi><mo separator="true">,</mo><mi>x</mi><mo separator="true">,</mo><mi>y</mi><mo stretchy="false">)</mo><mo>=</mo><mn>1</mn><mo>−</mo><munder><mo>∏</mo><mrow><mi>v</mi><mo>∈</mo><msub><mi>A</mi><mrow><mi>x</mi><mi mathvariant="normal">/</mi><mi>y</mi></mrow></msub></mrow></munder><mo stretchy="false">(</mo><mn>1</mn><mo>−</mo><msub><mi>b</mi><mi>v</mi></msub><mo stretchy="false">(</mo><mi>t</mi><mo separator="true">,</mo><mi>x</mi><mo separator="true">,</mo><mi>y</mi><mo stretchy="false">)</mo><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">\\ P_{Neut}(t,x,y)=1-\prod_{v\in A_{x/y}}(1-b_v(t,x,y))</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="mspace newline"></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3283em;"><span style="top:-2.55em;margin-left:-0.1389em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.10903em;">N</span><span class="mord mathnormal mtight">e</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord mathnormal">t</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">x</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">y</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:0.7278em;vertical-align:-0.0833em;"></span><span class="mord">1</span><span class="mspace" style="margin-right:0.2222em;"></span><span class="mbin">−</span><span class="mspace" style="margin-right:0.2222em;"></span></span><span class="base"><span class="strut" style="height:2.603em;vertical-align:-1.553em;"></span><span class="mop op-limits"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.05em;"><span style="top:-1.8557em;margin-left:0em;"><span class="pstrut" style="height:3.05em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.03588em;">v</span><span class="mrel mtight">∈</span><span class="mord mtight"><span class="mord mathnormal mtight">A</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3448em;"><span style="top:-2.3448em;margin-left:0em;margin-right:0.0714em;"><span class="pstrut" style="height:2.5357em;"></span><span class="sizing reset-size3 size1 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">x</span><span class="mord mtight">/</span><span class="mord mathnormal mtight" style="margin-right:0.03588em;">y</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.3695em;"><span></span></span></span></span></span></span></span></span></span><span style="top:-3.05em;"><span class="pstrut" style="height:3.05em;"></span><span><span class="mop op-symbol large-op">∏</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.553em;"><span></span></span></span></span></span><span class="mopen">(</span><span class="mord">1</span><span class="mspace" style="margin-right:0.2222em;"></span><span class="mbin">−</span><span class="mspace" style="margin-right:0.2222em;"></span></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathnormal">b</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.1514em;"><span style="top:-2.55em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.03588em;">v</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord mathnormal">t</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">x</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">y</span><span class="mclose">))</span></span></span></span></span></div>
<p>To compute this probability the binding probability <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>b</mi><mi>v</mi></msub><mo stretchy="false">(</mo><mi>t</mi><mo separator="true">,</mo><mi>x</mi><mo separator="true">,</mo><mi>y</mi><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">b_v(t,x,y)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathnormal">b</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.1514em;"><span style="top:-2.55em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.03588em;">v</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord mathnormal">t</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">x</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">y</span><span class="mclose">)</span></span></span></span></span> of an antibody of a particular epitope class with the variant is needed.</p>
<div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mspace linebreak="newline"></mspace><msub><mi>b</mi><mi>v</mi></msub><mo stretchy="false">(</mo><mi>t</mi><mo separator="true">,</mo><mi>x</mi><mo separator="true">,</mo><mi>y</mi><mo stretchy="false">)</mo><mo>=</mo><mfrac><mrow><mi>e</mi><mi>x</mi><msup><mi>p</mi><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>t</mi><mo>−</mo><mi>s</mi><mo stretchy="false">]</mo></mrow></msup></mrow><mrow><mi>F</mi><msub><mi>R</mi><mrow><mi>x</mi><mo separator="true">,</mo><mi>y</mi></mrow></msub><mo stretchy="false">(</mo><mi>v</mi><mo stretchy="false">)</mo><mi mathvariant="normal">.</mi><mi>I</mi><mi>C</mi><mn>5</mn><msub><mn>0</mn><mi>x</mi></msub><mo stretchy="false">(</mo><mi>v</mi><mo stretchy="false">)</mo><mo>+</mo><mi>e</mi><mi>x</mi><msup><mi>p</mi><mrow><mo>−</mo><mi>k</mi><mo stretchy="false">[</mo><mi>t</mi><mo>−</mo><mi>s</mi><mo stretchy="false">]</mo></mrow></msup></mrow></mfrac></mrow><annotation encoding="application/x-tex">\\ b_v(t,x,y)=\frac{exp^{-k[t-s]}}{FR_{x,y}(v).IC50_x(v)+exp^{-k[t-s]}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="mspace newline"></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathnormal">b</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.1514em;"><span style="top:-2.55em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.03588em;">v</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord mathnormal">t</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal">x</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">y</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:2.5551em;vertical-align:-0.9901em;"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.565em;"><span style="top:-2.296em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.13889em;">F</span><span class="mord"><span class="mord mathnormal" style="margin-right:0.00773em;">R</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.1514em;"><span style="top:-2.55em;margin-left:-0.0077em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">x</span><span class="mpunct mtight">,</span><span class="mord mathnormal mtight" style="margin-right:0.03588em;">y</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2861em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord mathnormal" style="margin-right:0.03588em;">v</span><span class="mclose">)</span><span class="mord">.</span><span class="mord mathnormal" style="margin-right:0.07847em;">I</span><span class="mord mathnormal" style="margin-right:0.07153em;">C</span><span class="mord">5</span><span class="mord"><span class="mord">0</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.1514em;"><span style="top:-2.55em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">x</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord mathnormal" style="margin-right:0.03588em;">v</span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222em;"></span><span class="mord mathnormal">e</span><span class="mord mathnormal">x</span><span class="mord"><span class="mord mathnormal">p</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.814em;"><span style="top:-2.989em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">t</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">s</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.677em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord mathnormal">e</span><span class="mord mathnormal">x</span><span class="mord"><span class="mord mathnormal">p</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.888em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">−</span><span class="mord mathnormal mtight" style="margin-right:0.03148em;">k</span><span class="mopen mtight">[</span><span class="mord mathnormal mtight">t</span><span class="mbin mtight">−</span><span class="mord mathnormal mtight">s</span><span class="mclose mtight">]</span></span></span></span></span></span></span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.9901em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></div>
<ul>
<li>The discount factor is used here to represent the diminishing concentration of the antibodies elicited by variant x. <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>F</mi><msub><mi>R</mi><mrow><mi>x</mi><mo separator="true">,</mo><mi>y</mi></mrow></msub><mo stretchy="false">(</mo><mi>v</mi><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">FR_{x,y}(v)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.0361em;vertical-align:-0.2861em;"></span><span class="mord mathnormal" style="margin-right:0.13889em;">F</span><span class="mord"><span class="mord mathnormal" style="margin-right:0.00773em;">R</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.1514em;"><span style="top:-2.55em;margin-left:-0.0077em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">x</span><span class="mpunct mtight">,</span><span class="mord mathnormal mtight" style="margin-right:0.03588em;">y</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2861em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord mathnormal" style="margin-right:0.03588em;">v</span><span class="mclose">)</span></span></span></span></span> is the fold resistance of a variant y to binding of antibodies of epitope class <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false">(</mo><mi>v</mi><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">(v)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mopen">(</span><span class="mord mathnormal" style="margin-right:0.03588em;">v</span><span class="mclose">)</span></span></span></span></span> elicited by variant x [All taken from the lab's paper which is currently in review].</li>
<li>The probability of binding is computed in the same manner as the pressure. For each postion that is getting mutated, for each of the antibodies the fold resistance and IC50 that was provided was used. These values were obtained from DMS data and was used in the predecessor work of the lab.</li>
<li>To compute the binding probability the FR and IC50 of antibodies belonging to the same epitope class was averaged and then used to compute the binding probability for an epitope class.</li>
<li>The resulting dataframe has the position, epitope class, <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>b</mi><mi>v</mi></msub></mrow><annotation encoding="application/x-tex">b_v</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8444em;vertical-align:-0.15em;"></span><span class="mord"><span class="mord mathnormal">b</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.1514em;"><span style="top:-2.55em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:0.03588em;">v</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span></span></span></span></span> series(for all the days in the observation time line). Binding proability with FR=1 and FR provided in the table were used to comput two resulting dataframes.</li>
<li>With the computed two binding probability dataframes, two probability of neutralisation dataframes are computed. <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>P</mi><msub><mn>1</mn><mrow><mi>N</mi><mi>e</mi><mi>u</mi><mi>t</mi></mrow></msub></msub></mrow><annotation encoding="application/x-tex">P_{1_{Neut}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.9336em;vertical-align:-0.2503em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3011em;"><span style="top:-2.55em;margin-left:-0.1389em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight"><span class="mord mtight">1</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3448em;"><span style="top:-2.3567em;margin-left:0em;margin-right:0.0714em;"><span class="pstrut" style="height:2.5em;"></span><span class="sizing reset-size3 size1 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.10903em;">N</span><span class="mord mathnormal mtight">e</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.1433em;"><span></span></span></span></span></span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2503em;"><span></span></span></span></span></span></span></span></span></span></span> (with FR=1), <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>P</mi><msub><mn>2</mn><mrow><mi>N</mi><mi>e</mi><mi>u</mi><mi>t</mi></mrow></msub></msub></mrow><annotation encoding="application/x-tex">P_{2_{Neut}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.9336em;vertical-align:-0.2503em;"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3011em;"><span style="top:-2.55em;margin-left:-0.1389em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight"><span class="mord mtight">2</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3448em;"><span style="top:-2.3567em;margin-left:0em;margin-right:0.0714em;"><span class="pstrut" style="height:2.5em;"></span><span class="sizing reset-size3 size1 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.10903em;">N</span><span class="mord mathnormal mtight">e</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">t</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.1433em;"><span></span></span></span></span></span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2503em;"><span></span></span></span></span></span></span></span></span></span></span>. The Probability neutralisation dataframes contains a column for the position, and other columns dedicated for all the time steps.</li>
<li>To compute the probability of neutralisation, for each position, for each time step, the binding probability of all the antibody classes for the time step are considered.</li>
<li>The weight is computed by <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mfrac><mrow><msub><mi>P</mi><msub><mn>1</mn><mrow><mi>N</mi><mi>e</mi><mi>u</mi><mi>t</mi></mrow></msub></msub><mtext> </mtext><mo stretchy="false">(</mo><mi>w</mi><mi>i</mi><mi>t</mi><mi>h</mi><mi>F</mi><mi>R</mi><mo>=</mo><mn>1</mn><mo stretchy="false">)</mo></mrow><msub><mi>P</mi><msub><mn>2</mn><mrow><mi>N</mi><mi>e</mi><mi>u</mi><mi>t</mi></mrow></msub></msub></mfrac></mrow><annotation encoding="application/x-tex">\frac{P_{1_{Neut}}\space (with FR=1)}{ P_{2_{Neut}}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.7187em;vertical-align:-0.6144em;"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.1044em;"><span style="top:-2.655em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3173em;"><span style="top:-2.357em;margin-left:-0.1389em;margin-right:0.0714em;"><span class="pstrut" style="height:2.5em;"></span><span class="sizing reset-size3 size1 mtight"><span class="mord mtight"><span class="mord mtight"><span class="mord mtight">2</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3448em;"><span style="top:-2.3448em;margin-left:0em;margin-right:0.1em;"><span class="pstrut" style="height:2.6833em;"></span><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.10903em;">N</span><span class="mord mathnormal mtight">e</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">t</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.3385em;"><span></span></span></span></span></span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.3848em;"><span></span></span></span></span></span></span></span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.5794em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.13889em;">P</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3173em;"><span style="top:-2.357em;margin-left:-0.1389em;margin-right:0.0714em;"><span class="pstrut" style="height:2.5em;"></span><span class="sizing reset-size3 size1 mtight"><span class="mord mtight"><span class="mord mtight"><span class="mord mtight">1</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3448em;"><span style="top:-2.3448em;margin-left:0em;margin-right:0.1em;"><span class="pstrut" style="height:2.6833em;"></span><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:0.10903em;">N</span><span class="mord mathnormal mtight">e</span><span class="mord mathnormal mtight">u</span><span class="mord mathnormal mtight">t</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.3385em;"><span></span></span></span></span></span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.3848em;"><span></span></span></span></span></span></span><span class="mspace mtight"><span class="mtight"> </span></span><span class="mopen mtight">(</span><span class="mord mathnormal mtight" style="margin-right:0.02691em;">w</span><span class="mord mathnormal mtight">i</span><span class="mord mathnormal mtight">t</span><span class="mord mathnormal mtight">h</span><span class="mord mathnormal mtight" style="margin-right:0.00773em;">FR</span><span class="mrel mtight">=</span><span class="mord mtight">1</span><span class="mclose mtight">)</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.6144em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span>.</li>
<li>The computed weight is then multiplied with the pressure timestepwise and then the weighted pressure is plotted - Positionwise and countrywise; all the positions are considered irrespective of being exposed or burried. <code>Work/Data_Analysis/pressure_plots/pos_pressure_trend_avg_wtpressure.pdf</code>, <code>Work/Data_Analysis/pressure_plots/pressure_trend_avg_wtpressure.pdf</code></li>
<li><code>Work/Data_Analysis/Big_goal.Rmd</code>, <code>Work/Data_Analysis/neutralisation_probability.Rmd</code> has all the scripts regarding the big goal.</li>
</ul>
<blockquote>
<p>Another method: binding probability for all the time steps for all the antibodies were computed first and then averaging these binding probability of antibodies across an antibody class was done and used to find the probability of neutralisation. The weight values did not differ drasitically to be precise there are only 116 values that differ. Just sticking to the earlier method.
<a href="https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html">Definition Reference</a></p>
</blockquote>
<h2 id="analysis-of-the-weighted-plot">Analysis of the weighted plot<a aria-hidden="true" class="anchor-heading icon-link" href="#analysis-of-the-weighted-plot"></a></h2>
<ul>
<li>
<p><strong>Overall trend across all the countries:</strong></p>
<blockquote>
<p>The plots Pictures represents the overall trend for the grouped positions</p>
</blockquote>
<p> <strong>Grouping positions by similar pressure trends</strong></p>
<p> The similar trend positions were visualised to see if, the are very near to experience same trend of pressure.</p>
<ul>
<li>
<p>373,142,501,19,440,505,478,371,417,376,339,408,375,405,498,477, 484  <img src="/SARS-Cov2/assets/plots/Pressure_rep_curve_plateau.png" alt="plateau curve">
<img src="/SARS-Cov2/assets/protein_locations/first_grp.png" alt="first group"></p>
<blockquote>
<p>position 19 is missing in the structure</p>
</blockquote>
</li>
<li>
<p>496,143,145 <img src="/SARS-Cov2/assets/plots/Sharp_increase_decrease.png" alt="Sharp curve">
<img src="/SARS-Cov2/assets/protein_locations/second_grp.png" alt="second group"></p>
<blockquote>
<p>position 145 is missing in the structure</p>
</blockquote>
</li>
<li>
<p>493 the trend is like the previous but with an offset in the peak <img src="/SARS-Cov2/assets/plots/pp_493.png" alt="pos 493">
<img src="/SARS-Cov2/assets/protein_locations/pos_493.png" alt="493"></p>
<blockquote>
<p>This position is 8.1<span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mover accent="true"><mi>A</mi><mo>˚</mo></mover></mrow><annotation encoding="application/x-tex">\AA</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.9468em;"></span><span class="mord accent"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.9468em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="mord mathnormal">A</span></span><span style="top:-3.2523em;"><span class="pstrut" style="height:3em;"></span><span class="accent-body" style="left:-0.2361em;"><span class="mord">˚</span></span></span></span></span></span></span></span></span></span></span> away from pos 496</p>
</blockquote>
</li>
<li>
<p>446,346 <img src="/SARS-Cov2/assets/plots/small_peak_big_curve.png" alt="small peak and big curve">
<img src="/SARS-Cov2/assets/protein_locations/446_346.png" alt="446_346"></p>
<blockquote>
<p>pos 446 is missing in the structure</p>
</blockquote>
</li>
<li>
<p>157,452 <img src="/SARS-Cov2/assets/plots/small_peak_big_curve_decrease.png" alt="small peak and big bump with decrease">
<img src="/SARS-Cov2/assets/protein_locations/452_157.png" alt="157_452"></p>
</li>
<li>
<p>158,156 <img src="/SARS-Cov2/assets/plots/small_peak_instability_end.png" alt="small peak and instability end">
<img src="/SARS-Cov2/assets/protein_locations/156_158.png" alt="156_156"></p>
</li>
<li>
<p>144 similar to the previous but many countries have trends of high pressure at the end of the observation timeline <img src="/SARS-Cov2/assets/plots/pp_144.png" alt="pos 144"></p>
<blockquote>
<p>position 144 not found in the structure</p>
</blockquote>
</li>
<li>
<p>245,153,450,18 <img src="/SARS-Cov2/assets/plots/fluctuating_peaks.png" alt="sharp fluctuating peaks">
<img src="/SARS-Cov2/assets/protein_locations/245_153_450_18.png" alt="this one"></p>
<blockquote>
<p>pos 18,153 are absent in the struct</p>
</blockquote>
</li>
<li>
<p>146,445,368,252,490,460,486 <img src="/SARS-Cov2/assets/plots/increase_end.png" alt="increase only in end">
<img src="/SARS-Cov2/assets/protein_locations/end_increase.png" alt="end increase"></p>
<blockquote>
<p>146, 252, 445 are absent in the structure</p>
</blockquote>
</li>
<li>
<p>152,147,257,356,444 <img src="/SARS-Cov2/assets/plots/bump_end.png" alt="end bump">
<img src="/SARS-Cov2/assets/protein_locations/end_bump.png" alt="end bump"></p>
<blockquote>
<p>152, 147, 257 are absent in the structure</p>
</blockquote>
</li>
<li>
<p>253,521 <img src="/SARS-Cov2/assets/plots/end_small_rise.png" alt="end small curve">
<img src="/SARS-Cov2/assets/protein_locations/253_521.png" alt="253_521"></p>
<blockquote>
<p>pos 253 is absent in the structure</p>
</blockquote>
</li>
<li>
<p>455,456 <img src="/SARS-Cov2/assets/plots/end_sharp_increase.png" alt="sharp increase">
<img src="/SARS-Cov2/assets/protein_locations/455_456.png" alt="455_456"></p>
</li>
</ul>
</li>
<li>
<p>Visually the positions that have similar pressure trends are not always in proximity on the folded protein. Since this is a static structure, there might be dynamics which can bring these positions together but this is just an hypothesis.</p>
</li>
<li>
<p>256 high pressure is observed in only australia at the end of the observation period <img src="/SARS-Cov2/assets/plots/pp_256.png" alt="pos 256"></p>
</li>
<li>
<p>344,494 Germany has higher pressure than others <img src="/SARS-Cov2/assets/plots/pp_344.png" alt="pos 344"> <img src="/SARS-Cov2/assets/plots/pp_494.png" alt="pos 494"></p>
</li>
<li>
<p>248 India and Norway records high pressure comparing to others <img src="/SARS-Cov2/assets/plots/pp_248.png" alt="pos 248"></p>
</li>
</ul>
<h2 id="comparison-of-pressure-trend-with-the-variant-trend">Comparison of pressure trend with the variant trend<a aria-hidden="true" class="anchor-heading icon-link" href="#comparison-of-pressure-trend-with-the-variant-trend"></a></h2>
<ul>
<li>To do this the characteristic mutations of each variants were obtained using the <strong>getMutationsByLineage</strong> function from outbreakinfo package. From these the positions were alone taken.</li>
<li>But set of positions are not exclusive for a variant, there is major overlap. That is positions in first group belongs to the characteristics mutation of almost all the variants. So based on the mutation position pressure trend its difficult to draw a corelation with variant trend.</li>
<li>Considering mutations would be a better way to go ahead.</li>
<li>I think considering individual mutations would give allow us to visualise occurance of mutation on a position after the pressure peaks on a position</li>
</ul>
<h2 id="repeating-the-process-with-mutations">Repeating the process with Mutations<a aria-hidden="true" class="anchor-heading icon-link" href="#repeating-the-process-with-mutations"></a></h2>
<ul>
<li>From frequency calculation to computing the weighted pressure the process was repeated again for individual mutations.</li>
<li>The visualization is slightly different. For each country there is a separate pdf. The pressure trends of mutations on the same positions are plotted together in a graph with each line representing a mutation.</li>
<li>Few trends are promising.</li>
<li><img src="/SARS-Cov2/assets/plots/india_promising.png" alt="India promising"> This seems like at the time when the pressure trend of spike_G339O peaks - high vulnerability to be mutated the pressure trend of spike_G339H starts picking up, this corroborates the hypothesis that when the pressure increases there are higher chances for that position to be mutated.</li>
<li><img src="/SARS-Cov2/assets/plots/australia_promising.png" alt="Australia promising"> In this case the mutant Spike_F157del has exsisted  along side of Spike_F157L. But the pressure on the former increases only after the pressure on the later peaks. This can be interpretted as the former got more favorable conditions to spread more and hence the pressure is going higher.</li>
<li>But there are trends that needs little more effort to look into <img src="/SARS-Cov2/assets/plots/plot_doubt1.png" alt="plot1"> <img src="/SARS-Cov2/assets/plots/plot_doubt2.png" alt="plot2"> <img src="/SARS-Cov2/assets/plots/plot_doubt3.png" alt="plot3"></li>
</ul>
<hr>
<strong>Children</strong>
<ol>
<li><a href="/SARS-Cov2/notes/g9181ia6dhnmpqfklf2mwit">Finding_surface_residues</a></li>
<li><a href="/SARS-Cov2/notes/vhn93xdheb909udqychwrcl">Mapping_lineages</a></li>
<li><a href="/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg">Possible_questions</a></li>
<li><a href="/SARS-Cov2/notes/o7n9flqgbqf3xbm437bc0ih">checking inconsistencies</a></li>
</ol>
<hr>
<strong>Backlinks</strong>
<ul>
<li><a href="/SARS-Cov2/notes/r423m96u71ix4pb458fk8u2">Work_documented</a></li>
</ul></div></div><div style="padding-left:10px;padding-right:10px" class="ant-col ant-col-xs-0 ant-col-md-6"><div><div class=""><div class="ant-anchor-wrapper dendron-toc" style="max-height:calc(100vh - 64px);z-index:1"><div class="ant-anchor"><div class="ant-anchor-ink"><span class="ant-anchor-ink-ball"></span></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#objective" title="Objective">Objective</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#steps-in-the-workflow" title="Steps in the workflow">Steps in the workflow</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#data-acquisition" title="Data acquisition">Data acquisition</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#summary" title="Summary">Summary</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#mapping-lineages" title="Mapping lineages">Mapping lineages</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#frequency-computation" title="Frequency computation">Frequency computation</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#dealing-with-the-unassigned" title="Dealing with the unassigned">Dealing with the unassigned</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#finding-positions-under-pressure-big-goal" title="Finding positions under pressure (BIG GOAL)">Finding positions under pressure (BIG GOAL)</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#computing-the-pressure-on-the-position" title="Computing the pressure on the position">Computing the pressure on the position</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#masking" title="Masking">Masking</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#visualization-of-the-positions-under-pressure" title="Visualization of the positions under pressure">Visualization of the positions under pressure</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#entropy-for-each-positions" title="Entropy for each positions:">Entropy for each positions:</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#neutralisation-probability" title="Neutralisation probability:">Neutralisation probability:</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#analysis-of-the-weighted-plot" title="Analysis of the weighted plot">Analysis of the weighted plot</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#comparison-of-pressure-trend-with-the-variant-trend" title="Comparison of pressure trend with the variant trend">Comparison of pressure trend with the variant trend</a></div><div class="ant-anchor-link"><a class="ant-anchor-link-title" href="#repeating-the-process-with-mutations" title="Repeating the process with Mutations">Repeating the process with Mutations</a></div></div></div></div></div></div></div></div></div></div></div><div class="ant-divider ant-divider-horizontal" role="separator"></div><footer class="ant-layout-footer" style="padding:0 24px 24px"></footer></main></section></section></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{"note":{"id":"r423m96u71ix4pb458fk8u2","title":"Work_documented","desc":"This is file contains all the steps done for the master thesis","updated":1710924226157,"created":1700240700998,"custom":{},"fname":"Work_documented","type":"note","vault":{"fsPath":".","selfContained":true,"name":"Literature"},"contentHash":"23d717c73a4d2f44e19241b397366581","links":[{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Work_documented.possible_questions","alias":"Question on date","position":{"start":{"line":12,"column":3,"offset":464},"end":{"line":12,"column":165,"offset":626},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Work_documented.possible_questions","anchorHeader":"1-whyhow-was-this-particular-time-period-jan22-oct23-chosen-while-pandemic-has-been-prevelant-2020-onwards"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Glossary","alias":"Hampler Filter","position":{"start":{"line":23,"column":135,"offset":1908},"end":{"line":23,"column":177,"offset":1950},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Glossary","anchorHeader":"hampler-filter"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Glossary","alias":"Outlier Estimation and methods","position":{"start":{"line":23,"column":190,"offset":1963},"end":{"line":23,"column":264,"offset":2037},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Glossary","anchorHeader":"outlier-estimation-and-methods"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Work_documented.checking inconsistencies","position":{"start":{"line":34,"column":234,"offset":4041},"end":{"line":34,"column":278,"offset":4085},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Work_documented.checking inconsistencies"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Work_documented","alias":"Note on the unassigned","position":{"start":{"line":51,"column":1,"offset":6518},"end":{"line":51,"column":71,"offset":6588},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Work_documented","anchorHeader":"dealing-with-the-unassigned"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Work_documented.possible_questions","alias":"Question on the treshold","position":{"start":{"line":72,"column":224,"offset":7764},"end":{"line":72,"column":376,"offset":7916},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Work_documented.possible_questions","anchorHeader":"3-the-jaccard-index-treshold-chosen-to-decide-a-parent-or-a-neighbour-is-50-is-this-okay"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Glossary","alias":"Confidence interval","position":{"start":{"line":94,"column":375,"offset":10252},"end":{"line":94,"column":427,"offset":10304},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Glossary","anchorHeader":"confidence-interval"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Analysing_variant_trends","position":{"start":{"line":96,"column":281,"offset":11286},"end":{"line":96,"column":309,"offset":11314},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Analysing_variant_trends"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Analysing_variant_trends","position":{"start":{"line":98,"column":73,"offset":11844},"end":{"line":98,"column":101,"offset":11872},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Analysing_variant_trends"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Work_documented.possible_questions","alias":"Question on the position","position":{"start":{"line":122,"column":249,"offset":14231},"end":{"line":122,"column":374,"offset":14356},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Work_documented.possible_questions","anchorHeader":"5-aaccording-to-uniprot-the-rbd-region-in-spike--is-319-541aa"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Glossary","alias":"interpolation","position":{"start":{"line":135,"column":81,"offset":15564},"end":{"line":135,"column":121,"offset":15604},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Glossary","anchorHeader":"interpolation"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Work_documented.possible_questions","alias":"Question on interpolation","position":{"start":{"line":137,"column":1,"offset":15789},"end":{"line":137,"column":130,"offset":15918},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Work_documented.possible_questions","anchorHeader":"4-why-do-we-do-linear-interpolation-why-not-spline-interpolation"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Work_documented.possible_questions","alias":"Question on solvent acessibility","position":{"start":{"line":161,"column":115,"offset":17682},"end":{"line":161,"column":454,"offset":18021},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Work_documented.possible_questions","anchorHeader":"6-if-a-rbd-spike-position-in-the-wildtype-is-occupied-by-a-hydrophobic-residue-and-it-is-replaced-by-hydrophilic-residue-the-solvent-accessibility-might-change-probably-due-to-the-difference-in-the-fold--in-that-case-should-we-study-these-positions-in-each-of-the-voi"}},{"type":"wiki","from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"value":"Work_documented.Finding_surface_residues","position":{"start":{"line":162,"column":100,"offset":18121},"end":{"line":162,"column":144,"offset":18165},"indent":[]},"xvault":false,"sameFile":false,"to":{"fname":"Work_documented.Finding_surface_residues"}},{"from":{"fname":"Work_documented","id":"r423m96u71ix4pb458fk8u2","vaultName":"Literature"},"type":"backlink","position":{"start":{"line":51,"column":1,"offset":6518},"end":{"line":51,"column":71,"offset":6588},"indent":[]},"value":"Work_documented"}],"anchors":{"objective":{"type":"header","text":"Objective","value":"objective","line":7,"column":0,"depth":1},"steps-in-the-workflow":{"type":"header","text":"Steps in the workflow","value":"steps-in-the-workflow","line":11,"column":0,"depth":2},"data-acquisition":{"type":"header","text":"Data acquisition","value":"data-acquisition","line":20,"column":0,"depth":2},"summary":{"type":"header","text":"Summary","value":"summary","line":41,"column":0,"depth":2},"mapping-lineages":{"type":"header","text":"Mapping lineages","value":"mapping-lineages","line":59,"column":0,"depth":2},"frequency-computation":{"type":"header","text":"Frequency computation","value":"frequency-computation","line":98,"column":0,"depth":2},"dealing-with-the-unassigned":{"type":"header","text":"Dealing with the unassigned","value":"dealing-with-the-unassigned","line":106,"column":0,"depth":2},"finding-positions-under-pressure-big-goal":{"type":"header","text":"Finding positions under pressure (BIG GOAL)","value":"finding-positions-under-pressure-big-goal","line":126,"column":0,"depth":2},"computing-the-pressure-on-the-position":{"type":"header","text":"Computing the pressure on the position","value":"computing-the-pressure-on-the-position","line":145,"column":0,"depth":2},"masking":{"type":"header","text":"Masking","value":"masking","line":164,"column":0,"depth":2},"visualization-of-the-positions-under-pressure":{"type":"header","text":"Visualization of the positions under pressure","value":"visualization-of-the-positions-under-pressure","line":183,"column":0,"depth":2},"entropy-for-each-positions":{"type":"header","text":"Entropy for each positions:","value":"entropy-for-each-positions","line":187,"column":0,"depth":2},"neutralisation-probability":{"type":"header","text":"Neutralisation probability:","value":"neutralisation-probability","line":191,"column":0,"depth":2},"analysis-of-the-weighted-plot":{"type":"header","text":"Analysis of the weighted plot","value":"analysis-of-the-weighted-plot","line":217,"column":0,"depth":2},"comparison-of-pressure-trend-with-the-variant-trend":{"type":"header","text":"Comparison of pressure trend with the variant trend","value":"comparison-of-pressure-trend-with-the-variant-trend","line":274,"column":0,"depth":2},"repeating-the-process-with-mutations":{"type":"header","text":"Repeating the process with Mutations","value":"repeating-the-process-with-mutations","line":281,"column":0,"depth":2}},"children":["g9181ia6dhnmpqfklf2mwit","vhn93xdheb909udqychwrcl","o7n9flqgbqf3xbm437bc0ih","xhw6w5ghbjhkzbo7huxzhwg"],"parent":"paa0s59lp320n6q8rghycjw","data":{}},"body":"\u003ch1 id=\"work_documented\"\u003eWork_documented\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#work_documented\"\u003e\u003c/a\u003e\u003c/h1\u003e\n\u003ch1 id=\"objective\"\u003eObjective\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#objective\"\u003e\u003c/a\u003e\u003c/h1\u003e\n\u003cp\u003eIs it possible to predict the positions that are vulnerable for forth coming mutations.\u003c/p\u003e\n\u003ch2 id=\"steps-in-the-workflow\"\u003eSteps in the workflow\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#steps-in-the-workflow\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cdiv class=\"mermaid\"\u003e\n  graph TD;\n1[start]--Based on data quantity--\u003eA[Choosing countries for the study]--checking for extreme low value outliers--\u003eB[Downloading sequence and meta data from these countries for 22 months]--Mapping lineages to parental lineage--\u003eC[Finding the difference in the trend of variant frequencies in these countries]\n\u003c/div\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#1-whyhow-was-this-particular-time-period-jan22-oct23-chosen-while-pandemic-has-been-prevelant-2020-onwards\"\u003eQuestion on date\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2 id=\"data-acquisition\"\u003eData acquisition\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#data-acquisition\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003eList of locations available in GISAID was obtained from GISAIDR Git repo - \u003ca href=\"https://github.com/Wytamma/GISAIDR/blob/master/GISAID_LOCATIONS.txt\"\u003eGISAID_LOCATIONS.txt\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eThe following points/steps are coded in the file \u003ccode\u003eWork/Data_Analysis/Monthwise_data_submission.Rmd\u003c/code\u003e\n\u003cul\u003e\n\u003cli\u003eFrom the GISAID, the total number of \u003cstrong\u003e\u003cem\u003ehigh\u003c/em\u003e\u003c/strong\u003e, \u003cstrong\u003e\u003cem\u003ecoverge\u003c/em\u003e\u003c/strong\u003e completed sequence deposited by each country (200+) in the past 22 months(Jan2022-Oct2023) were retrieved in a month wise fasion using the R Package GISAIDR. The table is in \u003ccode\u003eWork/Data_Analysis/country_monthly_submission.csv\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003eCountries that had more than 10,000 sequences deposited in total in the past 22 months were alone considered further.\u003c/li\u003e\n\u003cli\u003eThis included 10 countries : India, South Korea, Denmark, Germany, Norway, Spain, United Kingdom, Canada, USA, Australia. \u003ccode\u003eWork/Data_Analysis/chosen_ten_country_submission.csv\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003eThe trend of the time series submission statistics of these countries were plotted. It is present in the file \u003ccode\u003eWork/Data_Analysis/plots/ten_Country_data_trend_plots.pdf\u003c/code\u003e\n\u003cimg src=\"/SARS-Cov2/assets/plots/Country_trend_plots.png\" alt=\"Trends in 10 countries\"\u003e\u003c/li\u003e\n\u003cli\u003eOutlier detection and pruning of the observation horizon(Time period considered): Extreme low values were searched for, using the \u003ca href=\"/SARS-Cov2/notes/v4t96j5kch5mq12272qsyih#hampler-filter\"\u003eHampler Filter\u003c/a\u003e (one of the \u003ca href=\"/SARS-Cov2/notes/v4t96j5kch5mq12272qsyih#outlier-estimation-and-methods\"\u003eOutlier Estimation and methods\u003c/a\u003e). outliers_mad from Routliers package was used . Since there were no extreme low values found in any of the countries, the observation time period was not pruned.\u003c/li\u003e\n\u003cli\u003eThe sequence data for 10 countries were downloaded. There are files with crude data - all data retrieved from GISAID for each of the accesssion ID (29 cols of data) and there are files with essential columns: strain,virus,collection_date,country,pangolineage,sequence, length of the sequence, GISAID accession_id. The crude data for 10 countries are present 10 files in the folder \u003ccode\u003eWork/Data_Analysis/ten_country_crude_data_seq\u003c/code\u003e and the essential column files are in the folder \u003ccode\u003eWork/Data_Analysis/ten_country_ess_data\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003eThe Rmd also has a codeblock for downloading data with AA_substitution. This data was needed for mapping lineages to their parental lineage. For this I used the modified download script(modified by me) \u003ccode\u003eWork/Data_Analysis/Modified_GISAIDRdownload.R\u003c/code\u003e.\n\u003cblockquote\u003e\n\u003cp\u003eThis can be done by \u003ccode\u003efix(Download)\u003c/code\u003e and modifying the script as we need\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003eThis mutation data are stored in \u003ccode\u003eWork/Data_Analysis/ten_country_mut_data\u003c/code\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp\u003eThere were some inconsistencies in the Date entries (only year no month or year and month an no date) such entries were removed.\nOn the course of doing this it was identified that the number of entries that was downloaded for each countries do not exactly match the numbers in \u003ccode\u003eWork/Data_Analysis/chosen_ten_country_submission.csv\u003c/code\u003e excluding few countries.\nAfter removing the inconsistent date entries the numbers(downloaded and number in file) for India match.\nThe inconsistencies are reported in the file \u003ccode\u003eWork/Data_Analysis/inconsistencies_in_data.numbers\u003c/code\u003e.\nThe number of downloaded entries are more than then numbers in file.\nAfter looking into it, it was observed that there were few more entries added to these countries in the time gap(around a week) between the compiling \u003ccode\u003eWork/Data_Analysis/chosen_ten_country_submission.csv\u003c/code\u003e and downloading data. \u003ca href=\"/SARS-Cov2/notes/o7n9flqgbqf3xbm437bc0ih\"\u003echecking inconsistencies\u003c/a\u003e\u003c/p\u003e\n\u003ch2 id=\"summary\"\u003eSummary\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#summary\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cp\u003eSummary of the downloaded data\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"table-responsive\"\u003e\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\u003ctable\u003e\u003cthead\u003e\u003ctr\u003e\u003cth\u003e\u003cstrong\u003eCountry\u003c/strong\u003e\u003c/th\u003e\u003cth\u003e\u003cstrong\u003eTotal number of entries downloaded (Jan22-Oct23)\u003c/strong\u003e\u003c/th\u003e\u003cth\u003e\u003cstrong\u003eEntries after removing the date inconsistencies\u003c/strong\u003e\u003c/th\u003e\u003cth\u003e\u003cstrong\u003eNumber of unassigned\u003c/strong\u003e\u003c/th\u003e\u003cth\u003e\u003cstrong\u003ePercentage of unassigned\u003c/strong\u003e\u003c/th\u003e\u003c/tr\u003e\u003c/thead\u003e\u003ctbody\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eAsia / India\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e18106\u003c/td\u003e\u003ctd\u003e15783\u003c/td\u003e\u003ctd\u003e0\u003c/td\u003e\u003ctd\u003e0 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eAsia / South Korea\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e13808\u003c/td\u003e\u003ctd\u003e13808\u003c/td\u003e\u003ctd\u003e0\u003c/td\u003e\u003ctd\u003e0 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eEurope / Denmark\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e163900\u003c/td\u003e\u003ctd\u003e163900\u003c/td\u003e\u003ctd\u003e7\u003c/td\u003e\u003ctd\u003e0,004 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eEurope / Germany\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e66374\u003c/td\u003e\u003ctd\u003e66236\u003c/td\u003e\u003ctd\u003e0\u003c/td\u003e\u003ctd\u003e0 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eEurope / Norway\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e12264\u003c/td\u003e\u003ctd\u003e12264\u003c/td\u003e\u003ctd\u003e0\u003c/td\u003e\u003ctd\u003e0 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eEurope / Spain\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e17991\u003c/td\u003e\u003ctd\u003e17921\u003c/td\u003e\u003ctd\u003e0\u003c/td\u003e\u003ctd\u003e0 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eEurope / United Kingdom\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e32835\u003c/td\u003e\u003ctd\u003e30597\u003c/td\u003e\u003ctd\u003e74\u003c/td\u003e\u003ctd\u003e0,242 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eNorth America / Canada\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e20791\u003c/td\u003e\u003ctd\u003e20791\u003c/td\u003e\u003ctd\u003e0\u003c/td\u003e\u003ctd\u003e0 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eNorth America / USA\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e75487\u003c/td\u003e\u003ctd\u003e74971\u003c/td\u003e\u003ctd\u003e5\u003c/td\u003e\u003ctd\u003e0,007 %\u003c/td\u003e\u003c/tr\u003e\u003ctr\u003e\u003ctd\u003e\u003cstrong\u003eOceania / Australia\u003c/strong\u003e\u003c/td\u003e\u003ctd\u003e13267\u003c/td\u003e\u003ctd\u003e13257\u003c/td\u003e\u003ctd\u003e5\u003c/td\u003e\u003ctd\u003e0,038 %\u003c/td\u003e\u003c/tr\u003e\u003c/tbody\u003e\u003c/table\u003e\u003c/div\u003e\n\u003cp\u003e\u003ca href=\"/SARS-Cov2/notes/r423m96u71ix4pb458fk8u2#dealing-with-the-unassigned\"\u003eNote on the unassigned\u003c/a\u003e\u003c/p\u003e\n\u003ch2 id=\"mapping-lineages\"\u003eMapping lineages\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#mapping-lineages\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eMapping of lineages to their parental lineage is done based on the Pangolin lineage naming string and the spike mutations:\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003eFrom the 10 files of \u003ccode\u003eWork/Data_Analysis/ten_country_mut_data\u003c/code\u003e the columns corresponding to the mutations and pango lineage were extracted and stored in a different file \u003ccode\u003eWork/Data_Analysis/ten_country_mut_data/ten_country_lineage_mut.csv\u003c/code\u003e.\nThis is done using awk command\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cpre class=\"language-bash\"\u003e\u003ccode class=\"language-Bash\"\u003e\u003cspan class=\"token function\"\u003eawk\u003c/span\u003e -F\u003cspan class=\"token string\"\u003e\";\"\u003c/span\u003e \u003cspan class=\"token string\"\u003e'{gsub(/[()]/,\"\",$6);print $5\";\"$6}'\u003c/span\u003e \nten_country_mut_data/* \u003cspan class=\"token punctuation\"\u003e\\\u003c/span\u003e\n\u003cspan class=\"token operator\"\u003e\u003e\u003c/span\u003eten_country_mut_data/ten_country_lineage_mut.csv\n\u003c/code\u003e\u003c/pre\u003e\n\u003cblockquote\u003e\n\u003cp\u003egsub is used to remove the brackets in the AA_mutations column.\u003cbr\u003e\nColumns of the CSV files are separated by a semicolon ';'\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eAll the following steps are coded in the Rmd file \u003ccode\u003eWork/Data_Analysis/Lineage_mapping.Rmd\u003c/code\u003e\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eAll the mutations pertaining to each of the lineage were combined.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eFrom this only the spike mutations were considered for the downstream work.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eJaccard index (intersections of sets/Union of sets). If the Jaccard value calculated  is less than 0.5 then the lineages being compared are either considered as parental or neighbour depending on their pangolin string. \u003ca href=\"/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#3-the-jaccard-index-treshold-chosen-to-decide-a-parent-or-a-neighbour-is-50-is-this-okay\"\u003eQuestion on the treshold\u003c/a\u003e\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003eThe Lineages were mapped to parental lineage based on the the calculated jaccard value which is based on the number of common spike mutations between pangolin lineages that have same names dropping the last character in the sublineage. VOC/VUM/VOI were explicitely mapped based on the list taken from GISAID.\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eFor Pangolin lineages that are considered as VOI/VUM/VOC, the  mapping is directly given in GISAID. This can be found in the file \u003ccode\u003eWork/Data_Analysis/GISAID_VOI_VOC_VOM_list.txt\u003c/code\u003e.\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eThis list was obtained by first downloading the Clade/Lineage,variants(tsv) from the GISAID Downloads prompt.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eThis tsv file has multiple columns. Interesting columns were type and Value. Using the following bash commands the list of GISAID_VOI_VOC_VOMlist.txt was compiled\u003c/p\u003e\n\u003cpre class=\"language-bash\"\u003e\u003ccode class=\"language-bash\"\u003e\u003cspan class=\"token function\"\u003eawk\u003c/span\u003e -F\u003cspan class=\"token punctuation\"\u003e\\\u003c/span\u003et \u003cspan class=\"token string\"\u003e'{if($3==\"Variant\") print $4}'\u003c/span\u003e \u003cspan class=\"token punctuation\"\u003e\\\u003c/span\u003e\ngisaid_variants_statistics_from_gisaid.tsv \u003cspan class=\"token operator\"\u003e|\u003c/span\u003e \u003cspan class=\"token punctuation\"\u003e\\\u003c/span\u003e\n\u003cspan class=\"token function\"\u003esort\u003c/span\u003e -u \u003cspan class=\"token operator\"\u003e\u003e\u003c/span\u003eGISAID_VOI_VOC_VOM_list.txt\n\u003c/code\u003e\u003c/pre\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eThe resulting text has long lines like \u003cem\u003e\u003cstrong\u003eVOI GRA (EG.5+EG.5.*) first detected in Indonesia/France\u003c/strong\u003e\u003c/em\u003e\nfrom this only the variant and sublineage names EG.5+EG.5.*  are retained while other texts including the brackets are removed(manually).\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eLines are rearranged in such a way that parental variant comes after the subvariant, example: XBB+XBB.*    would come only after XBB.1.16+XBB.1.16.*. This rearrangement was done manually.\n\u003cimg src=\"/SARS-Cov2/assets/Pics/Mapping_lineage_idea.png\" alt=\"mapping lineages idea\"\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eIn total there are 1628 unique lingeages which includes variants also. Running the mapping algorithm on this gives 715 lineages mapped to 16 variants and 913 lineages getting mapped to 531 lineages which includes Unknown.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp\u003eIf muation list is obtained using the outbreakinfo function getMutationsByLineage then the results vary for the non-variant lineages. 912(without unknown) lineages gets mapped to 676 lineages. This is present in assigning_the_unassigned.RMD\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2 id=\"frequency-computation\"\u003eFrequency computation\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#frequency-computation\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003eAfter the lineage mapping, for each country the frequency of presence of each parental_lineage(which also includes the sublienages) in a month was calculated (\u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmfrac\u003e\u003cmrow\u003e\u003cmi\u003ec\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003el\u003c/mi\u003e\u003cmi\u003ei\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003ea\u003c/mi\u003e\u003cmi\u003eg\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003eB\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003ei\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003ej\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003el\u003c/mi\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmi\u003ea\u003c/mi\u003e\u003cmi\u003el\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmi\u003er\u003c/mi\u003e\u003cmi\u003ei\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003ei\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003ej\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003el\u003c/mi\u003e\u003c/mrow\u003e\u003c/mfrac\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e\\frac{count\\space of\\space lineage\\space B\\space in\\space jul}{total\\space entries\\space in\\space jul}\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.4133em;vertical-align:-0.4811em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mopen nulldelimiter\"\u003e\u003c/span\u003e\u003cspan class=\"mfrac\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.9322em;\"\u003e\u003cspan style=\"top:-2.655em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eo\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ea\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.01968em;\"\u003el\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003en\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.02778em;\"\u003er\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ei\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ees\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ein\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.05724em;\"\u003ej\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.01968em;\"\u003el\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.23em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"frac-line\" style=\"border-bottom-width:0.04em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.4461em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eco\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003en\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eo\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.01968em;\"\u003el\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ein\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ea\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003eg\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.05017em;\"\u003eB\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ein\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.05724em;\"\u003ej\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.01968em;\"\u003el\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.4811em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mclose nulldelimiter\"\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e) with confidence interval and plotted.MultinomCI from DescTools package is used to compute the simultaneous \u003ca href=\"/SARS-Cov2/notes/v4t96j5kch5mq12272qsyih#confidence-interval\"\u003eConfidence interval\u003c/a\u003e. This function also calculates frequency/proportion so a seperate computation was not required.95% confidence interval with method sisonglaz was computed. The plots are stored in PDF file, 1 plot for each country so 10 pages - \u003ccode\u003eWork/Data_Analysis/plots/country_wise_frequency_trend.pdf\u003c/code\u003e without CI and \u003ccode\u003eWork/Data_Analysis/plots/country_wise_frequency_trend_CI.pdf\u003c/code\u003e with CI.\u003c/li\u003e\n\u003cli\u003eTo make it more convenient plots for each Parental_lineage/variant grouped by country was plotted to do the analysis. These are present in the pdf files - \u003ccode\u003eWork/Data_Analysis/plots/Variant_wise_trend_all_countries.pdf\u003c/code\u003e without CI and \u003ccode\u003eWork/Data_Analysis/plots/Variant_wise_trend_all_countries_CI.pdf\u003c/code\u003e with CI\u003c/li\u003e\n\u003cli\u003eThese plots are analysed to bring out the difference in the variant trend among countries. An entire spreedsheet was developed manually basically describing the variant trends in words \u003ccode\u003eWork/Data_Analysis/plot_Analysis_report.numbers\u003c/code\u003e, and the summary of the analysis is in \u003ca href=\"/SARS-Cov2/notes/1hnn8maoednnxdw5feqd4kq\"\u003eAnalysis of the varient trend plots\u003c/a\u003e file.\u003c/li\u003e\n\u003cli\u003eTo not miss on the difference of pandemic in countries the lineages which are not variants were also used for comparison like the variants. There are 300+ plots in the file each page corresponding to one lineage which is not a variants and the data is grouped by country. This is presetn in \u003ccode\u003eWork/Data_Analysis/plots/Lineages_no_var_freq_countrywise_CI\u003c/code\u003e with CI and in \u003ccode\u003eWork/Data_Analysis/plots/Lineages_no_var_freq_countrywise\u003c/code\u003e without CI.\u003c/li\u003e\n\u003cli\u003eFrom these plots trends that look interesting are chosen and noted in \u003ca href=\"/SARS-Cov2/notes/1hnn8maoednnxdw5feqd4kq\"\u003eAnalysis of the varient trend plots\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2 id=\"dealing-with-the-unassigned\"\u003eDealing with the unassigned\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#dealing-with-the-unassigned\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003eFor few entries of the GISAID downloaded data the lineage was not assigned. The lineage column had either a blank space or it was termed as unassigned. There were in total 91 unassigned entries in the downloaded 10 country data.\u003c/li\u003e\n\u003cli\u003eDenmark 7 entries ,UK 74 entries, USA 5 entries, Australia 5 which is Denmark 0.004%, UK 0.242%, USA 0.007%, Australia 0.038% of the entries in the respective countries.\u003c/li\u003e\n\u003cli\u003eMajority of the unassigned entries were collected in Oct23\u003c/li\u003e\n\u003cli\u003eTo get the possible assignment for these entries few methods were tried out.\u003c/li\u003e\n\u003cli\u003eFirst among them was to calculate distance between the unassigned and all other lineages based on their spike mutations. For this process the characteristics mutation for each lineage was obtained using the outbreakinfo function \u003cem\u003e\u003cstrong\u003egetMutationsByLineage\u003c/strong\u003e\u003c/em\u003e. Lineages that had less distance to each of the unassigned entries were considered as possible assignment.\u003c/li\u003e\n\u003cli\u003eThough this method seems to work when the method was checked with entries with known assignments, the method did not perform well.\u003c/li\u003e\n\u003cli\u003eThe main reason that this wouldn't work is beacuse not all the characteristic spike mutation is always present in the GISAID downloaded data, so when distance is computed there is always discrepencies.\u003c/li\u003e\n\u003cli\u003eHence to go with an accurate method genome sequences of these unassigned entries were obtained and a fasta file was compiled.\u003c/li\u003e\n\u003cli\u003eThis file \u003ccode\u003eunassigned_sequences.fasta\u003c/code\u003e was uploaded in the \u003ca href=\"https://pangolin.cog-uk.io\"\u003ehttps://pangolin.cog-uk.io\u003c/a\u003e to get the assignment.\u003c/li\u003e\n\u003cli\u003eOnce the assignment was obtained it was added to the dataset. This part is included in the fill_parental function of lineage_mapping. The trends were reanalysed after assigning the unassigned.\n\u003cblockquote\u003e\n\u003cp\u003eThe \u003ca href=\"https://pangolin.cog-uk.io\"\u003ehttps://pangolin.cog-uk.io\u003c/a\u003e was not able to analyse the 5 sequences from australia. No reasons was provided.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003e\u003cimg src=\"/SARS-Cov2/assets/Pics/pangolin_assignment_analysis.png\" alt=\"pangolin_assignment\"\u003e\u003c/p\u003e\n\u003cp\u003e\u003cimg src=\"/SARS-Cov2/assets/Pics/pangolin_assignment.png\" alt=\"pangolin_assignemt_details\"\u003e\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003eAll these are present in the \u003ccode\u003eWork/Data_Analysis/assigning_the_unassigned.Rmd\u003c/code\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2 id=\"finding-positions-under-pressure-big-goal\"\u003eFinding positions under pressure (BIG GOAL)\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#finding-positions-under-pressure-big-goal\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cp\u003eThe big goal is to find the positions under pressure. To obtain this, firstly the frequency of  mutation on each position of Spike RBD and NTD Epitopes are calculated. (RBD spike mutations in position 330-530, NTD(14-20,140-158,245-264) mutations)(\u003ca href=\"/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#5-aaccording-to-uniprot-the-rbd-region-in-spike--is-319-541aa\"\u003eQuestion on the position\u003c/a\u003e) For this the aa_substitution data obtained from GISAID was used.\n\u003cimg src=\"/SARS-Cov2/assets/Pics/Frequency_interpolation.png\" alt=\"frequency interpolation\"\u003e\u003c/p\u003e\n\u003cp\u003e\u003cstrong\u003eSTEPS:\u003c/strong\u003e\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003eFor Each country the mutation data from GISAID has been used as the input.\u003c/li\u003e\n\u003cli\u003eThis data consist of the lineage, collected date, location of collection, a string of aa_substitution.\u003c/li\u003e\n\u003cli\u003eAll the aa_substitution strings that are collected on the same day are considered together. This also gives the number of entries collected in a day = number sequences colleceted on a day.\u003c/li\u003e\n\u003cli\u003eFrom these strings the spike RBD, NTD epitope mutations are alone extracted using getintrmut function.\u003c/li\u003e\n\u003cli\u003eThe data frame gets reduced to date,number of sequences,spike RBD/NTD mutations string.\u003c/li\u003e\n\u003cli\u003eTo calculate the frequency of a mutation for a time step, all the possible unique spike RBD, NTD mutations present in the country is first compiled.\u003c/li\u003e\n\u003cli\u003eThen the frequency of each of this mutation for a day is calculated.\n\u003cblockquote\u003e\n\u003cp\u003eFrequency of pos_373 on 01-01-2022 = \u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmfrac\u003e\u003cmrow\u003e\u003cmi\u003ec\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003em\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmi\u003ea\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmi\u003ei\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003ei\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmi mathvariant=\"normal\"\u003e_\u003c/mi\u003e\u003cmn\u003e373\u003c/mn\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmn\u003e01\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmn\u003e01\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmn\u003e2022\u003c/mn\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmi\u003eN\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003em\u003c/mi\u003e\u003cmi\u003eb\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003er\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003eq\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmi\u003ec\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmn\u003e01\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmn\u003e01\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmn\u003e2022\u003c/mn\u003e\u003c/mrow\u003e\u003c/mfrac\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e\\frac{count\\space of\\space mutations\\space in\\space pos\\_373\\space on\\space 01-01-2022}{Number\\space of\\space sequences\\space on\\space 01-01-2022}\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.4942em;vertical-align:-0.4811em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mopen nulldelimiter\"\u003e\u003c/span\u003e\u003cspan class=\"mfrac\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:1.0131em;\"\u003e\u003cspan style=\"top:-2.655em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10903em;\"\u003eN\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003emb\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.02778em;\"\u003eer\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eo\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ese\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003eq\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003en\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eces\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eo\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003en\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e01\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e01\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e2022\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.23em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"frac-line\" style=\"border-bottom-width:0.04em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.527em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eco\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003en\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eo\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003em\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ea\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ei\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eo\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003en\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003es\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ein\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eos\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e_373\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eo\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003en\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e01\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e01\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e2022\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.4811em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mclose nulldelimiter\"\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003eTo have the frequency data for everyday from Jan 1 2022 - Oct 31 2023, linear \u003ca href=\"/SARS-Cov2/notes/v4t96j5kch5mq12272qsyih#interpolation\"\u003einterpolation\u003c/a\u003e is done using the approximate method. The result of the interpolation is added to the corresponding date in the country data_frame. This step is necessary to fill in the data gaps.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003e\u003ca href=\"/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#4-why-do-we-do-linear-interpolation-why-not-spline-interpolation\"\u003eQuestion on interpolation\u003c/a\u003e\u003c/p\u003e\n\u003ch2 id=\"computing-the-pressure-on-the-position\"\u003eComputing the pressure on the position\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#computing-the-pressure-on-the-position\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cp\u003eWith the formula (given by Prof.Max) the pressure on each position from t\u003csub\u003e0\u003c/sub\u003e to t was computed\n\u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmspace linebreak=\"newline\"\u003e\u003c/mspace\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e=\u003c/mo\u003e\u003cmsubsup\u003e\u003cmo\u003e∑\u003c/mo\u003e\u003cmrow\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo\u003e=\u003c/mo\u003e\u003cmsub\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmn\u003e0\u003c/mn\u003e\u003c/msub\u003e\u003c/mrow\u003e\u003cmi\u003et\u003c/mi\u003e\u003c/msubsup\u003e\u003cmsup\u003e\u003cmrow\u003e\u003cmi\u003eexp\u003c/mi\u003e\u003cmo\u003e⁡\u003c/mo\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003cmo\u003e×\u003c/mo\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmspace linebreak=\"newline\"\u003e\u003c/mspace\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e\\\\ P(pos,t)=\\sum_{s=t_0}^{t}\\exp^{-k[t-s]}\\times f(pos,s) \\\\\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"mspace newline\"\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003et\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003cspan class=\"mrel\"\u003e=\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.3333em;vertical-align:-0.3998em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop\"\u003e\u003cspan class=\"mop op-symbol small-op\" style=\"position:relative;top:0em;\"\u003e∑\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.9335em;\"\u003e\u003cspan style=\"top:-2.4003em;margin-left:0em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003es\u003c/span\u003e\u003cspan class=\"mrel mtight\"\u003e=\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3173em;\"\u003e\u003cspan style=\"top:-2.357em;margin-left:0em;margin-right:0.0714em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.5em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size3 size1 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e0\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.143em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.2029em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3998em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop\"\u003e\u003cspan class=\"mop\"\u003eexp\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.888em;\"\u003e\u003cspan style=\"top:-3.063em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003es\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e×\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003es\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace newline\"\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003eWhere the f(pos,t) is the frequency of the position on time t.\u003c/li\u003e\n\u003cli\u003e\u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmsup\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003eexp^{-k[t-s]}\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.0824em;vertical-align:-0.1944em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ex\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.888em;\"\u003e\u003cspan style=\"top:-3.063em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003es\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e is the discount factor - mutation frequencies that occurred [t-s] days ago get discounted by the half life of neutralising antibodies.\u003c/li\u003e\n\u003cli\u003e\u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo\u003e∼\u003c/mo\u003e\u003cmfrac\u003e\u003cmrow\u003e\u003cmi\u003el\u003c/mi\u003e\u003cmi\u003en\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmn\u003e45\u003c/mn\u003e\u003cmo\u003e+\u003c/mo\u003e\u003cmn\u003e14\u003c/mn\u003e\u003c/mrow\u003e\u003c/mfrac\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003ek\\sim \\frac{ln(2)}{45+14}\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:0.6944em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003cspan class=\"mrel\"\u003e∼\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.4133em;vertical-align:-0.4033em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mopen nulldelimiter\"\u003e\u003c/span\u003e\u003cspan class=\"mfrac\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:1.01em;\"\u003e\u003cspan style=\"top:-2.655em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e45\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e+\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e14\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.23em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"frac-line\" style=\"border-bottom-width:0.04em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.485em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.01968em;\"\u003el\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003en\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e2\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.4033em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mclose nulldelimiter\"\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/li\u003e\n\u003cli\u003eBy suggestion of the Prof. the vector for discount factor was first computed. For this the date range for each of the country_df was found. If the difference between the start day and the end day is 9 then [t-s] could be in the range 0-9. Hence with this as base the discount factor was computed for [t-s] ranging 0-[difference between the start day to end day in the dataframe]. All these values are stored in a vector.\u003c/li\u003e\n\u003cli\u003eFrom the start date to the end date the pressure was calculated for everday in the following manner\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"math math-display\"\u003e\u003cspan class=\"katex-display\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\" display=\"block\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmspace linebreak=\"newline\"\u003e\u003c/mspace\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e=\u003c/mo\u003e\u003cmsup\u003e\u003cmrow\u003e\u003cmi\u003eexp\u003c/mi\u003e\u003cmo\u003e⁡\u003c/mo\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003cmo\u003e×\u003c/mo\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmspace linebreak=\"newline\"\u003e\u003c/mspace\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e=\u003c/mo\u003e\u003cmsup\u003e\u003cmrow\u003e\u003cmi\u003eexp\u003c/mi\u003e\u003cmo\u003e⁡\u003c/mo\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003cmo\u003e×\u003c/mo\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e+\u003c/mo\u003e\u003cmsup\u003e\u003cmrow\u003e\u003cmi\u003eexp\u003c/mi\u003e\u003cmo\u003e⁡\u003c/mo\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003cmo\u003e×\u003c/mo\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmspace linebreak=\"newline\"\u003e\u003c/mspace\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e3\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e=\u003c/mo\u003e\u003cmsup\u003e\u003cmrow\u003e\u003cmi\u003eexp\u003c/mi\u003e\u003cmo\u003e⁡\u003c/mo\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e3\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003cmo\u003e×\u003c/mo\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e+\u003c/mo\u003e\u003cmsup\u003e\u003cmrow\u003e\u003cmi\u003eexp\u003c/mi\u003e\u003cmo\u003e⁡\u003c/mo\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e3\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003cmo\u003e×\u003c/mo\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e+\u003c/mo\u003e\u003cmsup\u003e\u003cmrow\u003e\u003cmi\u003eexp\u003c/mi\u003e\u003cmo\u003e⁡\u003c/mo\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e3\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e3\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003cmo\u003e×\u003c/mo\u003e\u003cmi\u003ef\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmi\u003eo\u003c/mi\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ed\u003c/mi\u003e\u003cmn\u003e3\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e\\\\ P(pos,d1)=\\exp^{-k[d1-d1]}\\times f(pos,d1) \\\\ P(pos,d2)=\\exp^{-k[d2-d1]}\\times f(pos,d1) + \\exp^{-k[d2-d2]}\\times f(pos,d2)\\\\ P(pos,d3)=\\exp^{-k[d3-d1]}\\times f(pos,d1) + \\exp^{-k[d3-d2]}\\times f(pos,d2) + \\exp^{-k[d3-d3]}\\times f(pos,d3)\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"mspace newline\"\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e1\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003cspan class=\"mrel\"\u003e=\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.188em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop\"\u003e\u003cspan class=\"mop\"\u003eexp\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.938em;\"\u003e\u003cspan style=\"top:-3.113em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e1\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e1\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e×\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e1\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace newline\"\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e2\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003cspan class=\"mrel\"\u003e=\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.188em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop\"\u003e\u003cspan class=\"mop\"\u003eexp\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.938em;\"\u003e\u003cspan style=\"top:-3.113em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e2\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e1\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e×\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e1\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003cspan class=\"mbin\"\u003e+\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.188em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop\"\u003e\u003cspan class=\"mop\"\u003eexp\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.938em;\"\u003e\u003cspan style=\"top:-3.113em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e2\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e2\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e×\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e2\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace newline\"\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e3\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003cspan class=\"mrel\"\u003e=\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.188em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop\"\u003e\u003cspan class=\"mop\"\u003eexp\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.938em;\"\u003e\u003cspan style=\"top:-3.113em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e3\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e1\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e×\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e1\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003cspan class=\"mbin\"\u003e+\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.188em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop\"\u003e\u003cspan class=\"mop\"\u003eexp\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.938em;\"\u003e\u003cspan style=\"top:-3.113em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e3\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e2\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e×\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e2\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003cspan class=\"mbin\"\u003e+\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.188em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop\"\u003e\u003cspan class=\"mop\"\u003eexp\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.938em;\"\u003e\u003cspan style=\"top:-3.113em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e3\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ed\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e3\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e×\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.10764em;\"\u003ef\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eos\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ed\u003c/span\u003e\u003cspan class=\"mord\"\u003e3\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003eThis was done for all the RBD, NTD positions in a country and repeated for all the 10 countries.\u003c/li\u003e\n\u003cli\u003eThe output will have one column with the RBD and NTD positions and with columns as many as the number of days with data in the country\n\u003cimg src=\"/SARS-Cov2/assets/Pics/Pressure_computed_india.png\" alt=\"Pressure output\"\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2 id=\"masking\"\u003eMasking\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#masking\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003eThe objective of masking to find the exposed positions among the RBD, NTD spike positions.\u003c/li\u003e\n\u003cli\u003eTo know the exposed residues the solvent accessibility of each of the residues in the spike protein were found. \u003ca href=\"/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg#6-if-a-rbd-spike-position-in-the-wildtype-is-occupied-by-a-hydrophobic-residue-and-it-is-replaced-by-hydrophilic-residue-the-solvent-accessibility-might-change-probably-due-to-the-difference-in-the-fold--in-that-case-should-we-study-these-positions-in-each-of-the-voi\"\u003eQuestion on solvent acessibility\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eTo find solvent accessibility of the protein various tools were utilitsed. This can be found here\u003ca href=\"/SARS-Cov2/notes/g9181ia6dhnmpqfklf2mwit\"\u003eFinding_surface_residues\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eFrom the output of each of the tool Spike RBD,NTD surface positions were found. This process was direct in the case of the output from GetArea and Netsurf3.0 but in the case of DSSP, relative solvent accessibility was computed from the absolute solvent accessibility in the dssp output file. Using this computed relative solvent accessibility the surface residues were found.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp\u003eThe categorising treshold is set as 25% similar to Netsurf3.0. This would need advice.\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003eThe number of surface residues flagged by the 3 tools vary.\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003eDssp : 74 residues, NetSur3.0 : 123 residues, GetArea : 50 residues. All three tools flags 36 positions in common, GetArea and netsurf flags 38 common positions, dssp-getarea 48 positions in common, dssp-netsurf 54 positions in common.\u003c/li\u003e\n\u003cli\u003eDssp : 12 NTD, 62 RBD\u003c/li\u003e\n\u003cli\u003eGetArea: 8 NTD, 42 RBD\u003c/li\u003e\n\u003cli\u003eNetsurf: 22 NTD, 101 RBD\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eResidues flagged by each of the tools are fished from the ten country's position under pressure dataframe. The numbers are tabulated .\u003cimg src=\"/SARS-Cov2/assets/Pics/methods_comparison.png\" alt=\"Methods comparison\"\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp\u003eThe positions being picked by each of the tools are compared with the positions given by Prof.Max(screenshot). The tabulation is available in \u003ccode\u003eWork/Data_Analysis/Positions_being_picked.numbers\u003c/code\u003e\u003c/p\u003e\n\u003ch2 id=\"visualization-of-the-positions-under-pressure\"\u003eVisualization of the positions under pressure\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#visualization-of-the-positions-under-pressure\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cp\u003eLike previous variant trend visualization, the pressure trend was also visualised using geom_line in two ways, positionwise and countrywise. These plots are present in \u003ccode\u003eWork/Data_Analysis/pressure_plots\u003c/code\u003e. There are 6 plots - 3 countrywise, 3 positionsise. The 3 denotes the 3 masking tool outputs that were used to generate the plots.\u003c/p\u003e\n\u003ch2 id=\"entropy-for-each-positions\"\u003eEntropy for each positions:\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#entropy-for-each-positions\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cp\u003eThe amino acid substitution on a position is not constant. Example: on position 14 mutations found in india include Spike_Q14H,Spike_Q14del,Spike_Q14R. So calculating the number of such prevailing mutations on a position is the entropy. This was calculated to weigh down the pressure on a postiion .... Not processed further yet\u003c/p\u003e\n\u003ch2 id=\"neutralisation-probability\"\u003eNeutralisation probability:\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#neutralisation-probability\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cp\u003eFor knowing how impactful would a mutation on a site be, the probaility of neutralisation is being calculated. The Probability of neutralisation of a variant y by the antibodies elicited by variant x at time t.\u003c/p\u003e\n\u003cdiv class=\"math math-display\"\u003e\u003cspan class=\"katex-display\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\" display=\"block\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmspace linebreak=\"newline\"\u003e\u003c/mspace\u003e\u003cmsub\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmrow\u003e\u003cmi\u003eN\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003c/mrow\u003e\u003c/msub\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ey\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e=\u003c/mo\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmunder\u003e\u003cmo\u003e∏\u003c/mo\u003e\u003cmrow\u003e\u003cmi\u003ev\u003c/mi\u003e\u003cmo\u003e∈\u003c/mo\u003e\u003cmsub\u003e\u003cmi\u003eA\u003c/mi\u003e\u003cmrow\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmi mathvariant=\"normal\"\u003e/\u003c/mi\u003e\u003cmi\u003ey\u003c/mi\u003e\u003c/mrow\u003e\u003c/msub\u003e\u003c/mrow\u003e\u003c/munder\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmsub\u003e\u003cmi\u003eb\u003c/mi\u003e\u003cmi\u003ev\u003c/mi\u003e\u003c/msub\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ey\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e\\\\ P_{Neut}(t,x,y)=1-\\prod_{v\\in A_{x/y}}(1-b_v(t,x,y))\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"mspace newline\"\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3283em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:-0.1389em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10903em;\"\u003eN\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.15em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003et\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ex\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03588em;\"\u003ey\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003cspan class=\"mrel\"\u003e=\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:0.7278em;vertical-align:-0.0833em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e1\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003cspan class=\"mbin\"\u003e−\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:2.603em;vertical-align:-1.553em;\"\u003e\u003c/span\u003e\u003cspan class=\"mop op-limits\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:1.05em;\"\u003e\u003cspan style=\"top:-1.8557em;margin-left:0em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3.05em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003cspan class=\"mrel mtight\"\u003e∈\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eA\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3448em;\"\u003e\u003cspan style=\"top:-2.3448em;margin-left:0em;margin-right:0.0714em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.5357em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size3 size1 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ex\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e/\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003ey\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3695em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3.05em;\"\u003e\u003c/span\u003e\u003cspan\u003e\u003cspan class=\"mop op-symbol large-op\"\u003e∏\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:1.553em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord\"\u003e1\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003cspan class=\"mbin\"\u003e−\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\"\u003eb\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1514em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:0em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.15em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003et\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ex\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03588em;\"\u003ey\u003c/span\u003e\u003cspan class=\"mclose\"\u003e))\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/div\u003e\n\u003cp\u003eTo compute this probability the binding probability \u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmsub\u003e\u003cmi\u003eb\u003c/mi\u003e\u003cmi\u003ev\u003c/mi\u003e\u003c/msub\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ey\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003eb_v(t,x,y)\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\"\u003eb\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1514em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:0em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.15em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003et\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ex\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03588em;\"\u003ey\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e of an antibody of a particular epitope class with the variant is needed.\u003c/p\u003e\n\u003cdiv class=\"math math-display\"\u003e\u003cspan class=\"katex-display\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\" display=\"block\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmspace linebreak=\"newline\"\u003e\u003c/mspace\u003e\u003cmsub\u003e\u003cmi\u003eb\u003c/mi\u003e\u003cmi\u003ev\u003c/mi\u003e\u003c/msub\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ey\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e=\u003c/mo\u003e\u003cmfrac\u003e\u003cmrow\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmsup\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003c/mrow\u003e\u003cmrow\u003e\u003cmi\u003eF\u003c/mi\u003e\u003cmsub\u003e\u003cmi\u003eR\u003c/mi\u003e\u003cmrow\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ey\u003c/mi\u003e\u003c/mrow\u003e\u003c/msub\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ev\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmi mathvariant=\"normal\"\u003e.\u003c/mi\u003e\u003cmi\u003eI\u003c/mi\u003e\u003cmi\u003eC\u003c/mi\u003e\u003cmn\u003e5\u003c/mn\u003e\u003cmsub\u003e\u003cmn\u003e0\u003c/mn\u003e\u003cmi\u003ex\u003c/mi\u003e\u003c/msub\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ev\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003cmo\u003e+\u003c/mo\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmsup\u003e\u003cmi\u003ep\u003c/mi\u003e\u003cmrow\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003ek\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e[\u003c/mo\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmo\u003e−\u003c/mo\u003e\u003cmi\u003es\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e]\u003c/mo\u003e\u003c/mrow\u003e\u003c/msup\u003e\u003c/mrow\u003e\u003c/mfrac\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e\\\\ b_v(t,x,y)=\\frac{exp^{-k[t-s]}}{FR_{x,y}(v).IC50_x(v)+exp^{-k[t-s]}}\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"mspace newline\"\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\"\u003eb\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1514em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:0em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.15em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003et\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ex\u003c/span\u003e\u003cspan class=\"mpunct\"\u003e,\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.1667em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03588em;\"\u003ey\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003cspan class=\"mrel\"\u003e=\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2778em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:2.5551em;vertical-align:-0.9901em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mopen nulldelimiter\"\u003e\u003c/span\u003e\u003cspan class=\"mfrac\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:1.565em;\"\u003e\u003cspan style=\"top:-2.296em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eF\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.00773em;\"\u003eR\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1514em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:-0.0077em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ex\u003c/span\u003e\u003cspan class=\"mpunct mtight\"\u003e,\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003ey\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.2861em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mord\"\u003e.\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.07847em;\"\u003eI\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.07153em;\"\u003eC\u003c/span\u003e\u003cspan class=\"mord\"\u003e5\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord\"\u003e0\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1514em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:0em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ex\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.15em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003cspan class=\"mbin\"\u003e+\u003c/span\u003e\u003cspan class=\"mspace\" style=\"margin-right:0.2222em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ex\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.814em;\"\u003e\u003cspan style=\"top:-2.989em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003es\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.23em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"frac-line\" style=\"border-bottom-width:0.04em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.677em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003ex\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\"\u003ep\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.888em;\"\u003e\u003cspan style=\"top:-3.063em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03148em;\"\u003ek\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e[\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mbin mtight\"\u003e−\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003es\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e]\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.9901em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mclose nulldelimiter\"\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/div\u003e\n\u003cul\u003e\n\u003cli\u003eThe discount factor is used here to represent the diminishing concentration of the antibodies elicited by variant x. \u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmi\u003eF\u003c/mi\u003e\u003cmsub\u003e\u003cmi\u003eR\u003c/mi\u003e\u003cmrow\u003e\u003cmi\u003ex\u003c/mi\u003e\u003cmo separator=\"true\"\u003e,\u003c/mo\u003e\u003cmi\u003ey\u003c/mi\u003e\u003c/mrow\u003e\u003c/msub\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ev\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003eFR_{x,y}(v)\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.0361em;vertical-align:-0.2861em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eF\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.00773em;\"\u003eR\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1514em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:-0.0077em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ex\u003c/span\u003e\u003cspan class=\"mpunct mtight\"\u003e,\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003ey\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.2861em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e is the fold resistance of a variant y to binding of antibodies of epitope class \u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ev\u003c/mi\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e(v)\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1em;vertical-align:-0.25em;\"\u003e\u003c/span\u003e\u003cspan class=\"mopen\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003cspan class=\"mclose\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e elicited by variant x [All taken from the lab's paper which is currently in review].\u003c/li\u003e\n\u003cli\u003eThe probability of binding is computed in the same manner as the pressure. For each postion that is getting mutated, for each of the antibodies the fold resistance and IC50 that was provided was used. These values were obtained from DMS data and was used in the predecessor work of the lab.\u003c/li\u003e\n\u003cli\u003eTo compute the binding probability the FR and IC50 of antibodies belonging to the same epitope class was averaged and then used to compute the binding probability for an epitope class.\u003c/li\u003e\n\u003cli\u003eThe resulting dataframe has the position, epitope class, \u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmsub\u003e\u003cmi\u003eb\u003c/mi\u003e\u003cmi\u003ev\u003c/mi\u003e\u003c/msub\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003eb_v\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:0.8444em;vertical-align:-0.15em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\"\u003eb\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1514em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:0em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.03588em;\"\u003ev\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.15em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e series(for all the days in the observation time line). Binding proability with FR=1 and FR provided in the table were used to comput two resulting dataframes.\u003c/li\u003e\n\u003cli\u003eWith the computed two binding probability dataframes, two probability of neutralisation dataframes are computed. \u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmsub\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmsub\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmrow\u003e\u003cmi\u003eN\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003c/mrow\u003e\u003c/msub\u003e\u003c/msub\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003eP_{1_{Neut}}\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:0.9336em;vertical-align:-0.2503em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3011em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:-0.1389em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e1\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3448em;\"\u003e\u003cspan style=\"top:-2.3567em;margin-left:0em;margin-right:0.0714em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.5em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size3 size1 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10903em;\"\u003eN\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1433em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.2503em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e (with FR=1), \u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmsub\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmsub\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmrow\u003e\u003cmi\u003eN\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003c/mrow\u003e\u003c/msub\u003e\u003c/msub\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003eP_{2_{Neut}}\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:0.9336em;vertical-align:-0.2503em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mord mathnormal\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3011em;\"\u003e\u003cspan style=\"top:-2.55em;margin-left:-0.1389em;margin-right:0.05em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.7em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e2\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3448em;\"\u003e\u003cspan style=\"top:-2.3567em;margin-left:0em;margin-right:0.0714em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.5em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size3 size1 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10903em;\"\u003eN\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.1433em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.2503em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e. The Probability neutralisation dataframes contains a column for the position, and other columns dedicated for all the time steps.\u003c/li\u003e\n\u003cli\u003eTo compute the probability of neutralisation, for each position, for each time step, the binding probability of all the antibody classes for the time step are considered.\u003c/li\u003e\n\u003cli\u003eThe weight is computed by \u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmfrac\u003e\u003cmrow\u003e\u003cmsub\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmsub\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmrow\u003e\u003cmi\u003eN\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003c/mrow\u003e\u003c/msub\u003e\u003c/msub\u003e\u003cmtext\u003e \u003c/mtext\u003e\u003cmo stretchy=\"false\"\u003e(\u003c/mo\u003e\u003cmi\u003ew\u003c/mi\u003e\u003cmi\u003ei\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003cmi\u003eh\u003c/mi\u003e\u003cmi\u003eF\u003c/mi\u003e\u003cmi\u003eR\u003c/mi\u003e\u003cmo\u003e=\u003c/mo\u003e\u003cmn\u003e1\u003c/mn\u003e\u003cmo stretchy=\"false\"\u003e)\u003c/mo\u003e\u003c/mrow\u003e\u003cmsub\u003e\u003cmi\u003eP\u003c/mi\u003e\u003cmsub\u003e\u003cmn\u003e2\u003c/mn\u003e\u003cmrow\u003e\u003cmi\u003eN\u003c/mi\u003e\u003cmi\u003ee\u003c/mi\u003e\u003cmi\u003eu\u003c/mi\u003e\u003cmi\u003et\u003c/mi\u003e\u003c/mrow\u003e\u003c/msub\u003e\u003c/msub\u003e\u003c/mfrac\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e\\frac{P_{1_{Neut}}\\space (with FR=1)}{ P_{2_{Neut}}}\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:1.7187em;vertical-align:-0.6144em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord\"\u003e\u003cspan class=\"mopen nulldelimiter\"\u003e\u003c/span\u003e\u003cspan class=\"mfrac\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:1.1044em;\"\u003e\u003cspan style=\"top:-2.655em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3173em;\"\u003e\u003cspan style=\"top:-2.357em;margin-left:-0.1389em;margin-right:0.0714em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.5em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size3 size1 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e2\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3448em;\"\u003e\u003cspan style=\"top:-2.3448em;margin-left:0em;margin-right:0.1em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.6833em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10903em;\"\u003eN\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3385em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3848em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.23em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"frac-line\" style=\"border-bottom-width:0.04em;\"\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.5794em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size6 size3 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.13889em;\"\u003eP\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3173em;\"\u003e\u003cspan style=\"top:-2.357em;margin-left:-0.1389em;margin-right:0.0714em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.5em;\"\u003e\u003c/span\u003e\u003cspan class=\"sizing reset-size3 size1 mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mtight\"\u003e1\u003c/span\u003e\u003cspan class=\"msupsub\"\u003e\u003cspan class=\"vlist-t vlist-t2\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3448em;\"\u003e\u003cspan style=\"top:-2.3448em;margin-left:0em;margin-right:0.1em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:2.6833em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.10903em;\"\u003eN\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ee\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eu\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3385em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.3848em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mspace mtight\"\u003e\u003cspan class=\"mtight\"\u003e \u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mopen mtight\"\u003e(\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.02691em;\"\u003ew\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003ei\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003et\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\"\u003eh\u003c/span\u003e\u003cspan class=\"mord mathnormal mtight\" style=\"margin-right:0.00773em;\"\u003eFR\u003c/span\u003e\u003cspan class=\"mrel mtight\"\u003e=\u003c/span\u003e\u003cspan class=\"mord mtight\"\u003e1\u003c/span\u003e\u003cspan class=\"mclose mtight\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-s\"\u003e​\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.6144em;\"\u003e\u003cspan\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"mclose nulldelimiter\"\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e.\u003c/li\u003e\n\u003cli\u003eThe computed weight is then multiplied with the pressure timestepwise and then the weighted pressure is plotted - Positionwise and countrywise; all the positions are considered irrespective of being exposed or burried. \u003ccode\u003eWork/Data_Analysis/pressure_plots/pos_pressure_trend_avg_wtpressure.pdf\u003c/code\u003e, \u003ccode\u003eWork/Data_Analysis/pressure_plots/pressure_trend_avg_wtpressure.pdf\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eWork/Data_Analysis/Big_goal.Rmd\u003c/code\u003e, \u003ccode\u003eWork/Data_Analysis/neutralisation_probability.Rmd\u003c/code\u003e has all the scripts regarding the big goal.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp\u003eAnother method: binding probability for all the time steps for all the antibodies were computed first and then averaging these binding probability of antibodies across an antibody class was done and used to find the probability of neutralisation. The weight values did not differ drasitically to be precise there are only 116 values that differ. Just sticking to the earlier method.\n\u003ca href=\"https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html\"\u003eDefinition Reference\u003c/a\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003ch2 id=\"analysis-of-the-weighted-plot\"\u003eAnalysis of the weighted plot\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#analysis-of-the-weighted-plot\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003e\u003cstrong\u003eOverall trend across all the countries:\u003c/strong\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003eThe plots Pictures represents the overall trend for the grouped positions\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp\u003e \u003cstrong\u003eGrouping positions by similar pressure trends\u003c/strong\u003e\u003c/p\u003e\n\u003cp\u003e The similar trend positions were visualised to see if, the are very near to experience same trend of pressure.\u003c/p\u003e\n\u003cul\u003e\n\u003cli\u003e\n\u003cp\u003e373,142,501,19,440,505,478,371,417,376,339,408,375,405,498,477, 484  \u003cimg src=\"/SARS-Cov2/assets/plots/Pressure_rep_curve_plateau.png\" alt=\"plateau curve\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/first_grp.png\" alt=\"first group\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003eposition 19 is missing in the structure\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e496,143,145 \u003cimg src=\"/SARS-Cov2/assets/plots/Sharp_increase_decrease.png\" alt=\"Sharp curve\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/second_grp.png\" alt=\"second group\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003eposition 145 is missing in the structure\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e493 the trend is like the previous but with an offset in the peak \u003cimg src=\"/SARS-Cov2/assets/plots/pp_493.png\" alt=\"pos 493\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/pos_493.png\" alt=\"493\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003eThis position is 8.1\u003cspan class=\"math math-inline\"\u003e\u003cspan class=\"katex\"\u003e\u003cspan class=\"katex-mathml\"\u003e\u003cmath xmlns=\"http://www.w3.org/1998/Math/MathML\"\u003e\u003csemantics\u003e\u003cmrow\u003e\u003cmover accent=\"true\"\u003e\u003cmi\u003eA\u003c/mi\u003e\u003cmo\u003e˚\u003c/mo\u003e\u003c/mover\u003e\u003c/mrow\u003e\u003cannotation encoding=\"application/x-tex\"\u003e\\AA\u003c/annotation\u003e\u003c/semantics\u003e\u003c/math\u003e\u003c/span\u003e\u003cspan class=\"katex-html\" aria-hidden=\"true\"\u003e\u003cspan class=\"base\"\u003e\u003cspan class=\"strut\" style=\"height:0.9468em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord accent\"\u003e\u003cspan class=\"vlist-t\"\u003e\u003cspan class=\"vlist-r\"\u003e\u003cspan class=\"vlist\" style=\"height:0.9468em;\"\u003e\u003cspan style=\"top:-3em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"mord mathnormal\"\u003eA\u003c/span\u003e\u003c/span\u003e\u003cspan style=\"top:-3.2523em;\"\u003e\u003cspan class=\"pstrut\" style=\"height:3em;\"\u003e\u003c/span\u003e\u003cspan class=\"accent-body\" style=\"left:-0.2361em;\"\u003e\u003cspan class=\"mord\"\u003e˚\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e away from pos 496\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e446,346 \u003cimg src=\"/SARS-Cov2/assets/plots/small_peak_big_curve.png\" alt=\"small peak and big curve\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/446_346.png\" alt=\"446_346\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003epos 446 is missing in the structure\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e157,452 \u003cimg src=\"/SARS-Cov2/assets/plots/small_peak_big_curve_decrease.png\" alt=\"small peak and big bump with decrease\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/452_157.png\" alt=\"157_452\"\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e158,156 \u003cimg src=\"/SARS-Cov2/assets/plots/small_peak_instability_end.png\" alt=\"small peak and instability end\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/156_158.png\" alt=\"156_156\"\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e144 similar to the previous but many countries have trends of high pressure at the end of the observation timeline \u003cimg src=\"/SARS-Cov2/assets/plots/pp_144.png\" alt=\"pos 144\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003eposition 144 not found in the structure\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e245,153,450,18 \u003cimg src=\"/SARS-Cov2/assets/plots/fluctuating_peaks.png\" alt=\"sharp fluctuating peaks\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/245_153_450_18.png\" alt=\"this one\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003epos 18,153 are absent in the struct\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e146,445,368,252,490,460,486 \u003cimg src=\"/SARS-Cov2/assets/plots/increase_end.png\" alt=\"increase only in end\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/end_increase.png\" alt=\"end increase\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003e146, 252, 445 are absent in the structure\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e152,147,257,356,444 \u003cimg src=\"/SARS-Cov2/assets/plots/bump_end.png\" alt=\"end bump\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/end_bump.png\" alt=\"end bump\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003e152, 147, 257 are absent in the structure\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e253,521 \u003cimg src=\"/SARS-Cov2/assets/plots/end_small_rise.png\" alt=\"end small curve\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/253_521.png\" alt=\"253_521\"\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp\u003epos 253 is absent in the structure\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e455,456 \u003cimg src=\"/SARS-Cov2/assets/plots/end_sharp_increase.png\" alt=\"sharp increase\"\u003e\n\u003cimg src=\"/SARS-Cov2/assets/protein_locations/455_456.png\" alt=\"455_456\"\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003eVisually the positions that have similar pressure trends are not always in proximity on the folded protein. Since this is a static structure, there might be dynamics which can bring these positions together but this is just an hypothesis.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e256 high pressure is observed in only australia at the end of the observation period \u003cimg src=\"/SARS-Cov2/assets/plots/pp_256.png\" alt=\"pos 256\"\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e344,494 Germany has higher pressure than others \u003cimg src=\"/SARS-Cov2/assets/plots/pp_344.png\" alt=\"pos 344\"\u003e \u003cimg src=\"/SARS-Cov2/assets/plots/pp_494.png\" alt=\"pos 494\"\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp\u003e248 India and Norway records high pressure comparing to others \u003cimg src=\"/SARS-Cov2/assets/plots/pp_248.png\" alt=\"pos 248\"\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2 id=\"comparison-of-pressure-trend-with-the-variant-trend\"\u003eComparison of pressure trend with the variant trend\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#comparison-of-pressure-trend-with-the-variant-trend\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003eTo do this the characteristic mutations of each variants were obtained using the \u003cstrong\u003egetMutationsByLineage\u003c/strong\u003e function from outbreakinfo package. From these the positions were alone taken.\u003c/li\u003e\n\u003cli\u003eBut set of positions are not exclusive for a variant, there is major overlap. That is positions in first group belongs to the characteristics mutation of almost all the variants. So based on the mutation position pressure trend its difficult to draw a corelation with variant trend.\u003c/li\u003e\n\u003cli\u003eConsidering mutations would be a better way to go ahead.\u003c/li\u003e\n\u003cli\u003eI think considering individual mutations would give allow us to visualise occurance of mutation on a position after the pressure peaks on a position\u003c/li\u003e\n\u003c/ul\u003e\n\u003ch2 id=\"repeating-the-process-with-mutations\"\u003eRepeating the process with Mutations\u003ca aria-hidden=\"true\" class=\"anchor-heading icon-link\" href=\"#repeating-the-process-with-mutations\"\u003e\u003c/a\u003e\u003c/h2\u003e\n\u003cul\u003e\n\u003cli\u003eFrom frequency calculation to computing the weighted pressure the process was repeated again for individual mutations.\u003c/li\u003e\n\u003cli\u003eThe visualization is slightly different. For each country there is a separate pdf. The pressure trends of mutations on the same positions are plotted together in a graph with each line representing a mutation.\u003c/li\u003e\n\u003cli\u003eFew trends are promising.\u003c/li\u003e\n\u003cli\u003e\u003cimg src=\"/SARS-Cov2/assets/plots/india_promising.png\" alt=\"India promising\"\u003e This seems like at the time when the pressure trend of spike_G339O peaks - high vulnerability to be mutated the pressure trend of spike_G339H starts picking up, this corroborates the hypothesis that when the pressure increases there are higher chances for that position to be mutated.\u003c/li\u003e\n\u003cli\u003e\u003cimg src=\"/SARS-Cov2/assets/plots/australia_promising.png\" alt=\"Australia promising\"\u003e In this case the mutant Spike_F157del has exsisted  along side of Spike_F157L. But the pressure on the former increases only after the pressure on the later peaks. This can be interpretted as the former got more favorable conditions to spread more and hence the pressure is going higher.\u003c/li\u003e\n\u003cli\u003eBut there are trends that needs little more effort to look into \u003cimg src=\"/SARS-Cov2/assets/plots/plot_doubt1.png\" alt=\"plot1\"\u003e \u003cimg src=\"/SARS-Cov2/assets/plots/plot_doubt2.png\" alt=\"plot2\"\u003e \u003cimg src=\"/SARS-Cov2/assets/plots/plot_doubt3.png\" alt=\"plot3\"\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003chr\u003e\n\u003cstrong\u003eChildren\u003c/strong\u003e\n\u003col\u003e\n\u003cli\u003e\u003ca href=\"/SARS-Cov2/notes/g9181ia6dhnmpqfklf2mwit\"\u003eFinding_surface_residues\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/SARS-Cov2/notes/vhn93xdheb909udqychwrcl\"\u003eMapping_lineages\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/SARS-Cov2/notes/xhw6w5ghbjhkzbo7huxzhwg\"\u003ePossible_questions\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"/SARS-Cov2/notes/o7n9flqgbqf3xbm437bc0ih\"\u003echecking inconsistencies\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003chr\u003e\n\u003cstrong\u003eBacklinks\u003c/strong\u003e\n\u003cul\u003e\n\u003cli\u003e\u003ca href=\"/SARS-Cov2/notes/r423m96u71ix4pb458fk8u2\"\u003eWork_documented\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e","noteIndex":{"id":"paa0s59lp320n6q8rghycjw","title":"Work on SARS Cov2","desc":"","updated":1700240746536,"created":1700233379150,"custom":{"nav_order":0,"permalink":"/"},"fname":"root","type":"note","vault":{"fsPath":".","selfContained":true,"name":"Literature"},"contentHash":"6124b7f80065618c049c13a49f0dbed1","links":[],"anchors":{},"children":["1hnn8maoednnxdw5feqd4kq","nve3ld8zkeg6b2hgygrh1rp","v4t96j5kch5mq12272qsyih","r423m96u71ix4pb458fk8u2"],"parent":null,"data":{},"body":"\n"},"collectionChildren":null,"customHeadContent":null,"config":{"version":5,"dev":{"enablePreviewV2":true,"enableSelfContainedVaults":true},"commands":{"lookup":{"note":{"selectionMode":"extract","confirmVaultOnCreate":true,"vaultSelectionModeOnCreate":"smart","leaveTrace":false,"bubbleUpCreateNew":true,"fuzzThreshold":0.2}},"randomNote":{},"insertNoteLink":{"aliasMode":"none","enableMultiSelect":false},"insertNoteIndex":{"enableMarker":false},"copyNoteLink":{"aliasMode":"title"},"templateHierarchy":"template"},"workspace":{"vaults":[{"fsPath":".","selfContained":true,"name":"Literature"}],"journal":{"dailyDomain":"daily","name":"journal","dateFormat":"y.MM.dd","addBehavior":"childOfDomain"},"scratch":{"name":"scratch","dateFormat":"y.MM.dd.HHmmss","addBehavior":"asOwnDomain"},"task":{"name":"task","dateFormat":"y.MM.dd","addBehavior":"asOwnDomain","statusSymbols":{"":" ","wip":"w","done":"x","assigned":"a","moved":"m","blocked":"b","delegated":"l","dropped":"d","pending":"y"},"taskCompleteStatus":["done","x"],"prioritySymbols":{"H":"high","M":"medium","L":"low"},"todoIntegration":false,"createTaskSelectionType":"selection2link"},"graph":{"zoomSpeed":1,"createStub":false},"enableAutoCreateOnDefinition":false,"enableXVaultWikiLink":false,"enableRemoteVaultInit":true,"enableUserTags":true,"enableHashTags":true,"workspaceVaultSyncMode":"noCommit","enableAutoFoldFrontmatter":false,"enableEditorDecorations":true,"maxPreviewsCached":10,"maxNoteLength":204800,"enableFullHierarchyNoteTitle":false},"preview":{"enableFMTitle":true,"enableNoteTitleForLink":true,"enableFrontmatterTags":true,"enableHashesForFMTags":false,"enablePrettyRefs":true,"enableKatex":true,"automaticallyShowPreview":false},"publishing":{"enableFMTitle":true,"enableNoteTitleForLink":true,"enablePrettyRefs":true,"enableKatex":true,"copyAssets":true,"siteHierarchies":["root"],"writeStubs":false,"siteRootDir":"docs","seo":{"title":"Dendron","description":"Personal Knowledge Space"},"github":{"enableEditLink":true,"editLinkText":"Edit this page on GitHub","editBranch":"main","editViewMode":"tree"},"enableSiteLastModified":true,"enableFrontmatterTags":true,"enableHashesForFMTags":false,"enableRandomlyColoredTags":true,"enableTaskNotes":true,"enablePrettyLinks":true,"searchMode":"search","siteUrl":"https://vishnushiri02.github.io","assetsPrefix":"/SARS-Cov2","siteFaviconPath":"favicon.ico","siteIndex":"root"}}},"__N_SSG":true},"page":"/notes/[id]","query":{"id":"r423m96u71ix4pb458fk8u2"},"buildId":"npK0PGR6Fa7mwLej5tvS3","assetPrefix":"/SARS-Cov2","isFallback":false,"gsp":true,"scriptLoader":[]}</script></body></html>